<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The goal of ggtranscript is the simplify the process of visualizing 
    transcript structure and annotation. To achieve this, ggtranscript 
    introduces 5 new geoms (geom_range(), geom_half_range(), geom_intron(), 
    geom_junction() and geom_junction_label_repel()) as well as several helper 
    functions. As a ggplot2 extension, ggtranscript inherits ggplot2s 
    familiarity and flexibility, enabling users to intuitively adjust 
    aesthetics, parameters, scales etc as well as complement ggtranscript geoms
    with existing ggplot2 geoms to create informative, publication-ready 
    plots.">
<title>Visualizing Transcript Structure and Annotation using ggplot2 â€¢ ggtranscript</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Visualizing Transcript Structure and Annotation using ggplot2">
<meta property="og:description" content="The goal of ggtranscript is the simplify the process of visualizing 
    transcript structure and annotation. To achieve this, ggtranscript 
    introduces 5 new geoms (geom_range(), geom_half_range(), geom_intron(), 
    geom_junction() and geom_junction_label_repel()) as well as several helper 
    functions. As a ggplot2 extension, ggtranscript inherits ggplot2s 
    familiarity and flexibility, enabling users to intuitively adjust 
    aesthetics, parameters, scales etc as well as complement ggtranscript geoms
    with existing ggplot2 geoms to create informative, publication-ready 
    plots.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.9</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/ggtranscript.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="ggtranscript-">ggtranscript <img src="reference/figures/ggtranscript_logo_cropped.svg" align="right" height="139"><a class="anchor" aria-label="anchor" href="#ggtranscript-"></a>
</h1></div>
<!-- badges: start -->

<p><code>ggtranscript</code> is a <code>ggplot2</code> extension that makes it to easy to visualize transcript structure and annotation.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># you can install the development version of ggtranscript from GitHub:</span>
<span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"dzhang32/ggtranscript"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p><code>ggtranscript</code> introduces 5 new geoms (<code><a href="reference/geom_range.html">geom_range()</a></code>, <code><a href="reference/geom_range.html">geom_half_range()</a></code>, <code><a href="reference/geom_intron.html">geom_intron()</a></code>, <code><a href="reference/geom_junction.html">geom_junction()</a></code> and <code><a href="reference/geom_junction_label_repel.html">geom_junction_label_repel()</a></code>) and several helper functions designed to facilitate the visualization of transcript structure and annotation. The following guide takes you on a quick tour of using these geoms, for a more detailed overview see the <a href="https://dzhang32.github.io/ggtranscript/articles/ggtranscript.html" class="external-link">Getting Started tutorial</a>.</p>
<p><code><a href="reference/geom_range.html">geom_range()</a></code> and <code><a href="reference/geom_intron.html">geom_intron()</a></code> enable the plotting of exons and introns, the core components of transcript annotation. <code>ggtranscript</code> also provides <code><a href="reference/to_intron.html">to_intron()</a></code>, which converts exon co-ordinates to the corresponding introns. Together, <code>ggtranscript</code> enables users to plot transcript structures with only exons as the required input and just a few lines of code.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dzhang32/ggtranscript" class="external-link">ggtranscript</a></span><span class="op">)</span>

<span class="co"># to illustrate the package's functionality</span>
<span class="co"># ggtranscript includes example transcript annotation</span>
<span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 Ã— 8</span>
<span class="co">#&gt;   seqnames  start    end strand type  gene_name transcript_name transcript_biotâ€¦</span>
<span class="co">#&gt;   &lt;fct&gt;     &lt;int&gt;  &lt;int&gt; &lt;fct&gt;  &lt;fct&gt; &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;           </span>
<span class="co">#&gt; 1 21       3.17e7 3.17e7 +      gene  SOD1      &lt;NA&gt;            &lt;NA&gt;            </span>
<span class="co">#&gt; 2 21       3.17e7 3.17e7 +      tranâ€¦ SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; 3 21       3.17e7 3.17e7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; 4 21       3.17e7 3.17e7 +      CDS   SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; 5 21       3.17e7 3.17e7 +      starâ€¦ SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; 6 21       3.17e7 3.17e7 +      exon  SOD1      SOD1-202        protein_coding</span>

<span class="co"># extract exons</span>
<span class="va">sod1_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span>

<span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-geom-range-intron-1.png" width="100%"></p>
<p><code>ggtranscript</code> provides the helper function <code><a href="reference/shorten_gaps.html">shorten_gaps()</a></code>, which reduces the size of the gaps. <code><a href="reference/shorten_gaps.html">shorten_gaps()</a></code> then rescales the exon and intron co-ordinates to preserve the original exon alignment. This allows you to hone in the differences in the exonic structure, which can be particularly useful if the transcript has relatively long introns.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sod1_rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/shorten_gaps.html">shorten_gaps</a></span><span class="op">(</span>
  <span class="va">sod1_exons</span>, 
  <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, 
  group_var <span class="op">=</span> <span class="st">"transcript_name"</span>
  <span class="op">)</span>

<span class="va">sod1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">sod1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"intron"</span><span class="op">)</span>, 
        arrow.min.intron.length <span class="op">=</span> <span class="fl">200</span>
    <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-shorten-gaps-1.png" width="100%"></p>
<p><code><a href="reference/geom_range.html">geom_range()</a></code> can be used for any range-based genomic annotation. For example, when plotting protein-coding transcripts, users may find it helpful to visually distinguish the coding segments from UTRs.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># filter for only exons from protein coding transcripts</span>
<span class="va">sod1_exons_prot_cod</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span>

<span class="co"># obtain cds</span>
<span class="va">sod1_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span>

<span class="va">sod1_exons_prot_cod</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        fill <span class="op">=</span> <span class="st">"white"</span>,
        height <span class="op">=</span> <span class="fl">0.25</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">sod1_cds</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons_prot_cod</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>,
        arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span>,
    <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-geom-range-intron-w-cds-1.png" width="100%"></p>
<p><code><a href="reference/geom_range.html">geom_half_range()</a></code> takes advantage of the vertical symmetry of transcript annotation by plotting only half of a range on the top or bottom of a transcript structure. One use case of <code><a href="reference/geom_range.html">geom_half_range()</a></code> is to visualize the differences between transcript structure more clearly.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract exons and cds for the two transcripts to be compared</span>
<span class="va">sod1_201_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span>
<span class="va">sod1_201_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span>
<span class="va">sod1_202_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-202"</span><span class="op">)</span>
<span class="va">sod1_202_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-202"</span><span class="op">)</span>

<span class="va">sod1_201_202_plot</span> <span class="op">&lt;-</span> <span class="va">sod1_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="st">"SOD1-201/202"</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_half_range</a></span><span class="op">(</span>
        fill <span class="op">=</span> <span class="st">"white"</span>,
        height <span class="op">=</span> <span class="fl">0.125</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_half_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">sod1_201_cds</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_half_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">sod1_202_exons</span>,
        range.orientation <span class="op">=</span> <span class="st">"top"</span>,
        fill <span class="op">=</span> <span class="st">"white"</span>,
        height <span class="op">=</span> <span class="fl">0.125</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_half_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">sod1_202_cds</span>,
        range.orientation <span class="op">=</span> <span class="st">"top"</span>,
        fill <span class="op">=</span> <span class="st">"purple"</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_202_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>
    <span class="op">)</span>

<span class="va">sod1_201_202_plot</span></code></pre></div>
<p><img src="reference/figures/README-geom-half-range-1.png" width="100%"></p>
<p>As a <code>ggplot2</code> extension, <code>ggtranscript</code> inherits the the familiarity and functionality of <code>ggplot2</code>. For instance, by leveraging <code><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian()</a></code> users can zoom in on regions of interest.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sod1_201_202_plot</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">31659500</span>, <span class="fl">31660000</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-geom-half-range-zoomed-1.png" width="100%"></p>
<p><code><a href="reference/geom_junction.html">geom_junction()</a></code> enables to plotting of junction curves, which can be overlaid across transcript structures. <code><a href="reference/geom_junction_label_repel.html">geom_junction_label_repel()</a></code> adds a label to junction curves, which can often be useful to mark junctions with a metric of their usage such as read counts.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ggtranscript includes a set of example (unannotated) junctions</span>
<span class="co"># originating from GTEx and downloaded via the Bioconductor package snapcount</span>
<span class="va">sod1_junctions</span>
<span class="co">#&gt; # A tibble: 5 Ã— 5</span>
<span class="co">#&gt;   seqnames    start      end strand mean_count</span>
<span class="co">#&gt;   &lt;fct&gt;       &lt;int&gt;    &lt;int&gt; &lt;fct&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt; 1 chr21    31659787 31666448 +           0.463</span>
<span class="co">#&gt; 2 chr21    31659842 31660554 +           0.831</span>
<span class="co">#&gt; 3 chr21    31659842 31663794 +           0.316</span>
<span class="co">#&gt; 4 chr21    31659842 31667257 +           4.35 </span>
<span class="co">#&gt; 5 chr21    31660351 31663789 +           0.324</span>

<span class="co"># add transcript_name to junctions for plotting</span>
<span class="va">sod1_junctions</span> <span class="op">&lt;-</span> <span class="va">sod1_junctions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>transcript_name <span class="op">=</span> <span class="st">"SOD1-201"</span><span class="op">)</span>

<span class="va">sod1_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
    xstart <span class="op">=</span> <span class="va">start</span>,
    xend <span class="op">=</span> <span class="va">end</span>,
    y <span class="op">=</span> <span class="va">transcript_name</span>
  <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
    fill <span class="op">=</span> <span class="st">"white"</span>, 
    height <span class="op">=</span> <span class="fl">0.25</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_201_cds</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="reference/geom_junction.html">geom_junction</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_junctions</span>,
    junction.y.max <span class="op">=</span> <span class="fl">0.5</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="reference/geom_junction_label_repel.html">geom_junction_label_repel</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_junctions</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">mean_count</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
    junction.y.max <span class="op">=</span> <span class="fl">0.5</span>
  <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-geom-junction-1.png" width="100%"></p>
<p>Alternatively, users may prefer to map junction read counts to the thickness of the junction curves. As a <code>ggplot2</code> extension, this can be done intuitively by modifying the size <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code> of <code><a href="reference/geom_junction.html">geom_junction()</a></code>. In addition, by modifying <code>ggplot2</code> scales and themes, users can easily create informative, publication-ready plots.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sod1_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
    xstart <span class="op">=</span> <span class="va">start</span>,
    xend <span class="op">=</span> <span class="va">end</span>,
    y <span class="op">=</span> <span class="va">transcript_name</span>
  <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
    fill <span class="op">=</span> <span class="st">"white"</span>, 
    height <span class="op">=</span> <span class="fl">0.25</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_201_cds</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="reference/geom_junction.html">geom_junction</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_junctions</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">mean_count</span><span class="op">)</span>,
    junction.y.max <span class="op">=</span> <span class="fl">0.5</span>, 
    ncp <span class="op">=</span> <span class="fl">30</span>, 
    colour <span class="op">=</span> <span class="st">"purple"</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size_continuous</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">1</span><span class="op">)</span>, guide <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Genomic position (chr21)"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Transcript name"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-geom-junction-pub-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"ggtranscript"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Gustavsson EK, Zhang D, Reynolds RH, Garcia-Ruiz S, Ryten M (2022).</span>
<span class="co">#&gt; "ggtranscript: an R package for the visualization and interpretation of</span>
<span class="co">#&gt; transcript isoforms using ggplot2." _Bioinformatics_. doi:</span>
<span class="co">#&gt; 10.1093/bioinformatics/btac409 (URL:</span>
<span class="co">#&gt; https://doi.org/10.1093/bioinformatics/btac409), &lt;URL:</span>
<span class="co">#&gt; https://academic.oup.com/bioinformatics/article/38/15/3844/6617821&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{,</span>
<span class="co">#&gt;     title = {ggtranscript: an R package for the visualization and interpretation of transcript isoforms using ggplot2},</span>
<span class="co">#&gt;     author = {Emil K Gustavsson and David Zhang and Regina H Reynolds and Sonia Garcia-Ruiz and Mina Ryten},</span>
<span class="co">#&gt;     year = {2022},</span>
<span class="co">#&gt;     journal = {Bioinformatics},</span>
<span class="co">#&gt;     doi = {https://doi.org/10.1093/bioinformatics/btac409},</span>
<span class="co">#&gt;     url = {https://academic.oup.com/bioinformatics/article/38/15/3844/6617821},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="credits">Credits<a class="anchor" aria-label="anchor" href="#credits"></a>
</h2>
<ul>
<li>
<code>ggtranscript</code> was developed using <em><a href="https://bioconductor.org/packages/3.14/biocthis" class="external-link">biocthis</a></em>.</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/dzhang32/ggtranscript/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/dzhang32/ggtranscript/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ggtranscript</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>David Zhang <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2382-8460" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Emil Gustavsson <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-0541-7537" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/dzhang32/ggtranscript/issues" class="external-link"><img src="https://img.shields.io/github/issues/dzhang32/ggtranscript" alt="GitHub issues"></a></li>
<li><a href="https://github.com/dzhang32/ggtranscript/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/dzhang32/ggtranscript" alt="GitHub pulls"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/dzhang32/ggtranscript/actions" class="external-link"><img src="https://github.com/dzhang32/ggtranscript/workflows/R-CMD-check-bioc/badge.svg" alt="R-CMD-check-bioc"></a></li>
<li><a href="https://app.codecov.io/gh/dzhang32/ggtranscript?branch=main" class="external-link"><img src="https://codecov.io/gh/dzhang32/ggtranscript/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
