<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Given a set of exons (encompassing the CDS and UTRs) and cds regions,
add_utr() will calculate and add the corresponding UTR regions as ranges.
This can be useful when combined with shorten_gaps() to visualize
transcripts with long introns, whilst differentiating UTRs from CDS regions."><title>Add untranslated regions (UTRs) — add_utr • ggtranscript</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Add untranslated regions (UTRs) — add_utr"><meta property="og:description" content="Given a set of exons (encompassing the CDS and UTRs) and cds regions,
add_utr() will calculate and add the corresponding UTR regions as ranges.
This can be useful when combined with shorten_gaps() to visualize
transcripts with long introns, whilst differentiating UTRs from CDS regions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/ggtranscript.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Add untranslated regions (UTRs)</h1>
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/add_utr.R" class="external-link"><code>R/add_utr.R</code></a></small>
      <div class="d-none name"><code>add_utr.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Given a set of <code>exons</code> (encompassing the CDS and UTRs) and <code>cds</code> regions,
<code>add_utr()</code> will calculate and add the corresponding UTR regions as ranges.
This can be useful when combined with <code><a href="shorten_gaps.html">shorten_gaps()</a></code> to visualize
transcripts with long introns, whilst differentiating UTRs from CDS regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">add_utr</span><span class="op">(</span><span class="va">exons</span>, <span class="va">cds</span>, group_var <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>exons</dt>
<dd><p><code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> contains exons which can originate from multiple
transcripts differentiated by <code>group_var</code>.</p></dd>
<dt>cds</dt>
<dd><p><code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> contains coding sequence ranges for the transcripts
in <code>exons</code>.</p></dd>
<dt>group_var</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> if input data originates from more than 1
transcript, <code>group_var</code> must specify the column that differentiates
transcripts (e.g. "transcript_id").</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> contains differentiated CDS and UTR ranges.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The definition of the inputted <code>cds</code> regions are expected to range from the
beginning of the start codon to the end of the stop codon. Sometimes, for
example in the case of Ensembl, reference annotation will omit the stop
codons from the CDS definition. In such cases, users should manually ensure
that the <code>cds</code> includes both the start and stop codons.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># to illustrate the package's functionality</span></span>
<span class="r-in"><span class="co"># ggtranscript includes example transcript annotation</span></span>
<span class="r-in"><span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      gene  PKNOX1    <span style="color: #BB0000;">NA</span>              <span style="color: #BB0000;">NA</span>              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      tran… PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># extract exons</span></span>
<span class="r-in"><span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># extract cds</span></span>
<span class="r-in"><span class="va">pknox1_cds</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">pknox1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      CDS   PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      CDS   PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      CDS   PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      CDS   PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      CDS   PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      CDS   PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># the CDS definition originating from the Ensembl reference annotation</span></span>
<span class="r-in"><span class="co"># does not include the stop codon</span></span>
<span class="r-in"><span class="co"># we must incorporate the stop codons into the CDS manually</span></span>
<span class="r-in"><span class="co"># by adding 3 base pairs to the end of the CDS of each transcript</span></span>
<span class="r-in"><span class="va">pknox1_cds_w_stop</span> <span class="op">&lt;-</span> <span class="va">pknox1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">transcript_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span class="r-in">        end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">end</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">end</span><span class="op">)</span>, <span class="va">end</span> <span class="op">+</span> <span class="fl">3</span>, <span class="va">end</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># add_utr() adds ranges that represent the UTRs</span></span>
<span class="r-in"><span class="va">pknox1_cds_utr</span> <span class="op">&lt;-</span> <span class="fu">add_utr</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">pknox1_exons</span>,</span>
<span class="r-in">    <span class="va">pknox1_cds_w_stop</span>,</span>
<span class="r-in">    group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">pknox1_cds_utr</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 9</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames    start      end width strand transcript_name type  gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664   155 +      PKNOX1-203      UTR   <span style="color: #BB0000;">NA</span>       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432   107 +      PKNOX1-203      UTR   <span style="color: #BB0000;">NA</span>       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618   128 +      PKNOX1-203      UTR   <span style="color: #BB0000;">NA</span>       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238   171 +      PKNOX1-203      CDS   PKNOX1   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       43<span style="text-decoration: underline;">016</span>908 43<span style="text-decoration: underline;">017</span>007   100 +      PKNOX1-203      CDS   PKNOX1   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       43<span style="text-decoration: underline;">018</span>133 43<span style="text-decoration: underline;">018</span>230    98 +      PKNOX1-203      CDS   PKNOX1   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># this can be useful when combined with shorten_gaps()</span></span>
<span class="r-in"><span class="co"># to visualize transcripts with long introns whilst differentiating UTRs</span></span>
<span class="r-in"><span class="va">pknox1_cds_utr_rescaled</span> <span class="op">&lt;-</span></span>
<span class="r-in">    <span class="fu"><a href="shorten_gaps.html">shorten_gaps</a></span><span class="op">(</span></span>
<span class="r-in">        exons <span class="op">=</span> <span class="va">pknox1_cds_utr</span>,</span>
<span class="r-in">        introns <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_cds_utr</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span class="r-in">        group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">pknox1_cds_utr_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">pknox1_cds_utr_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"UTR"</span><span class="op">)</span>,</span>
<span class="r-in">        height <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span class="r-in">        fill <span class="op">=</span> <span class="st">"white"</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span></span>
<span class="r-in">            <span class="va">pknox1_cds_utr_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">!=</span> <span class="st">"intron"</span><span class="op">)</span>,</span>
<span class="r-in">            <span class="st">"transcript_name"</span></span>
<span class="r-in">        <span class="op">)</span>,</span>
<span class="r-in">        arrow.min.intron.length <span class="op">=</span> <span class="fl">110</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-plt img"><img src="add_utr-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

