<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="geom_range() and geom_half_range() draw tiles that are designed to
represent range-based genomic features, such as exons. In combination with
geom_intron(), these geoms form the core components for visualizing
transcript structures."><title>Plot genomic ranges — geom_range • ggtranscript</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot genomic ranges — geom_range"><meta property="og:description" content="geom_range() and geom_half_range() draw tiles that are designed to
represent range-based genomic features, such as exons. In combination with
geom_intron(), these geoms form the core components for visualizing
transcript structures."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.9</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/ggtranscript.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Plot genomic ranges</h1>
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/geom_range.R" class="external-link"><code>R/geom_range.R</code></a>, <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/geom_half_range.R" class="external-link"><code>R/geom_half_range.R</code></a></small>
      <div class="d-none name"><code>geom_range.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>geom_range()</code> and <code>geom_half_range()</code> draw tiles that are designed to
represent range-based genomic features, such as exons. In combination with
<code><a href="geom_intron.html">geom_intron()</a></code>, these geoms form the core components for visualizing
transcript structures.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">geom_range</span><span class="op">(</span>
  mapping <span class="op">=</span> <span class="cn">NULL</span>,
  data <span class="op">=</span> <span class="cn">NULL</span>,
  stat <span class="op">=</span> <span class="st">"identity"</span>,
  position <span class="op">=</span> <span class="st">"identity"</span>,
  <span class="va">...</span>,
  vjust <span class="op">=</span> <span class="cn">NULL</span>,
  linejoin <span class="op">=</span> <span class="st">"mitre"</span>,
  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,
  show.legend <span class="op">=</span> <span class="cn">NA</span>,
  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="fu">geom_half_range</span><span class="op">(</span>
  mapping <span class="op">=</span> <span class="cn">NULL</span>,
  data <span class="op">=</span> <span class="cn">NULL</span>,
  stat <span class="op">=</span> <span class="st">"identity"</span>,
  position <span class="op">=</span> <span class="st">"identity"</span>,
  <span class="va">...</span>,
  range.orientation <span class="op">=</span> <span class="st">"bottom"</span>,
  linejoin <span class="op">=</span> <span class="st">"mitre"</span>,
  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,
  show.legend <span class="op">=</span> <span class="cn">NA</span>,
  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mapping</dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code> or
<code><a href="https://ggplot2.tidyverse.org/reference/aes_.html" class="external-link">aes_()</a></code>. If specified and <code>inherit.aes = TRUE</code> (the
default), it is combined with the default mapping at the top level of the
plot. You must supply <code>mapping</code> if there is no plot mapping.</p></dd>


<dt>data</dt>
<dd><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="https://ggplot2.tidyverse.org/reference/fortify.html" class="external-link">fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></dd>


<dt>stat</dt>
<dd><p>The statistical transformation to use on the data for this
layer, as a string.</p></dd>


<dt>position</dt>
<dd><p>Position adjustment, either as a string, or the result of
a call to a position adjustment function.</p></dd>


<dt>...</dt>
<dd><p>Other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p></dd>


<dt>vjust</dt>
<dd><p>horizontal and vertical justification of the grob.  Each
justification value should be a number between 0 and 1.  Defaults to 0.5
for both, centering each pixel over its data location.</p></dd>


<dt>linejoin</dt>
<dd><p>Line join style (round, mitre, bevel).</p></dd>


<dt>na.rm</dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>


<dt>show.legend</dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>


<dt>inherit.aes</dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>


<dt>range.orientation</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> one of "top" or "bottom", specifying
where the half ranges will be plotted with respect to each transcript
(<code>y</code>).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>the return value of a <code>geom_*</code> function is not intended to be
directly handled by users. Therefore, <code>geom_*</code> functions should never be
executed in isolation, rather used in combination with a
<code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2::ggplot()</a></code> call.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>geom_range()</code> and <code>geom_half_range()</code> require the following <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>;
<code>xstart</code>, <code>xend</code> and <code>y</code> (e.g. transcript name). <code>geom_half_range()</code> takes
advantage of the vertical symmetry of transcript annotation by plotting only
half of a range on the top or bottom of a transcript structure. This can be
useful for comparing between two transcripts or free up plotting space for
other transcript annotations (e.g. <code><a href="geom_junction.html">geom_junction()</a></code>).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># to illustrate the package's functionality</span></span>
<span class="r-in"><span class="co"># ggtranscript includes example transcript annotation</span></span>
<span class="r-in"><span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      gene  SOD1      <span style="color: #BB0000;">NA</span>              <span style="color: #BB0000;">NA</span>              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      tran… SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      CDS   SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      star… SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># extract exons</span></span>
<span class="r-in"><span class="va">sod1_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-204        processed_trans…</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">base</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">    <span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># geom_range() is designed to visualise range-based annotation such as exons</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_range</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># geom_half_range() allows users to plot half ranges</span></span>
<span class="r-in"><span class="co"># on the top or bottom of the transcript</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_half_range</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-2.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># where the half ranges are plotted can be adjusted using range.orientation</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_half_range</span><span class="op">(</span>range.orientation <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-3.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># as a ggplot2 extension, ggtranscript geoms inherit the</span></span>
<span class="r-in"><span class="co"># the functionality from the parameters and aesthetics in ggplot2</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_range</span><span class="op">(</span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_name</span><span class="op">)</span>,</span>
<span class="r-in">    size <span class="op">=</span> <span class="fl">1</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-4.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># together, geom_range() and geom_intron() are designed to visualize</span></span>
<span class="r-in"><span class="co"># the core components of transcript annotation</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_range</span><span class="op">(</span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-5.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># for protein coding transcripts</span></span>
<span class="r-in"><span class="co"># geom_range() be useful for visualizing UTRs that lie outside of the CDS</span></span>
<span class="r-in"><span class="va">sod1_exons_prot_coding</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># extract cds</span></span>
<span class="r-in"><span class="va">sod1_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sod1_exons_prot_coding</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_range</span><span class="op">(</span></span>
<span class="r-in">        fill <span class="op">=</span> <span class="st">"white"</span>,</span>
<span class="r-in">        height <span class="op">=</span> <span class="fl">0.25</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_range</span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">sod1_cds</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons_prot_coding</span>, <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-6.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># geom_half_range() can be useful for comparing between two transcripts</span></span>
<span class="r-in"><span class="co"># enabling visualization of one transcript on the top, other on the bottom</span></span>
<span class="r-in"><span class="va">sod1_201_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sod1_201_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sod1_202_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-202"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sod1_202_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-202"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sod1_201_plot</span> <span class="op">&lt;-</span> <span class="va">sod1_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">        y <span class="op">=</span> <span class="st">"SOD1-201/202"</span></span>
<span class="r-in">    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_half_range</span><span class="op">(</span></span>
<span class="r-in">        fill <span class="op">=</span> <span class="st">"white"</span>,</span>
<span class="r-in">        height <span class="op">=</span> <span class="fl">0.125</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_half_range</span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">sod1_201_cds</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sod1_201_plot</span></span>
<span class="r-plt img"><img src="geom_range-7.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sod1_201_202_plot</span> <span class="op">&lt;-</span> <span class="va">sod1_201_plot</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_half_range</span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">sod1_202_exons</span>,</span>
<span class="r-in">        range.orientation <span class="op">=</span> <span class="st">"top"</span>,</span>
<span class="r-in">        fill <span class="op">=</span> <span class="st">"white"</span>,</span>
<span class="r-in">        height <span class="op">=</span> <span class="fl">0.125</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_half_range</span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">sod1_202_cds</span>,</span>
<span class="r-in">        range.orientation <span class="op">=</span> <span class="st">"top"</span>,</span>
<span class="r-in">        fill <span class="op">=</span> <span class="st">"purple"</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_202_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sod1_201_202_plot</span></span>
<span class="r-plt img"><img src="geom_range-8.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># leveraging existing ggplot2 functionality via e.g. coord_cartesian()</span></span>
<span class="r-in"><span class="co"># can be useful to zoom in on areas of interest</span></span>
<span class="r-in"><span class="va">sod1_201_202_plot</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">31659500</span>, <span class="fl">31660000</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_range-9.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer></div>

  

  

  </body></html>

