<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="geom_intron() draws horizontal lines with central arrows that are designed
to represent introns. In combination with geom_range()/geom_half_range(),
these geoms form the core components for visualizing transcript
annotation."><title>Plot intron lines with strand arrows — geom_intron • ggtranscript</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot intron lines with strand arrows — geom_intron"><meta property="og:description" content="geom_intron() draws horizontal lines with central arrows that are designed
to represent introns. In combination with geom_range()/geom_half_range(),
these geoms form the core components for visualizing transcript
annotation."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/ggtranscript.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Plot intron lines with strand arrows</h1>
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/geom_intron.R" class="external-link"><code>R/geom_intron.R</code></a></small>
      <div class="d-none name"><code>geom_intron.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>geom_intron()</code> draws horizontal lines with central arrows that are designed
to represent introns. In combination with <code><a href="geom_range.html">geom_range()</a></code>/<code><a href="geom_range.html">geom_half_range()</a></code>,
these geoms form the core components for visualizing transcript
annotation.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">geom_intron</span><span class="op">(</span>
  mapping <span class="op">=</span> <span class="cn">NULL</span>,
  data <span class="op">=</span> <span class="cn">NULL</span>,
  stat <span class="op">=</span> <span class="st">"identity"</span>,
  position <span class="op">=</span> <span class="st">"identity"</span>,
  <span class="va">...</span>,
  arrow <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></span><span class="op">(</span>ends <span class="op">=</span> <span class="st">"last"</span>, length <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="st">"inches"</span><span class="op">)</span><span class="op">)</span>,
  arrow.fill <span class="op">=</span> <span class="cn">NULL</span>,
  lineend <span class="op">=</span> <span class="st">"butt"</span>,
  linejoin <span class="op">=</span> <span class="st">"round"</span>,
  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,
  arrow.min.intron.length <span class="op">=</span> <span class="fl">0</span>,
  show.legend <span class="op">=</span> <span class="cn">NA</span>,
  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mapping</dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code> or
<code><a href="https://ggplot2.tidyverse.org/reference/aes_.html" class="external-link">aes_()</a></code>. If specified and <code>inherit.aes = TRUE</code> (the
default), it is combined with the default mapping at the top level of the
plot. You must supply <code>mapping</code> if there is no plot mapping.</p></dd>
<dt>data</dt>
<dd><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="https://ggplot2.tidyverse.org/reference/fortify.html" class="external-link">fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></dd>
<dt>stat</dt>
<dd><p>The statistical transformation to use on the data for this
layer, as a string.</p></dd>
<dt>position</dt>
<dd><p>Position adjustment, either as a string, or the result of
a call to a position adjustment function.</p></dd>
<dt>...</dt>
<dd><p>Other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p></dd>
<dt>arrow</dt>
<dd><p>specification for arrow heads, as created by arrow().</p></dd>
<dt>arrow.fill</dt>
<dd><p>fill colour to use for the arrow head (if closed). <code>NULL</code>
means use <code>colour</code> aesthetic.</p></dd>
<dt>lineend</dt>
<dd><p>Line end style (round, butt, square).</p></dd>
<dt>linejoin</dt>
<dd><p>Line join style (round, mitre, bevel).</p></dd>
<dt>na.rm</dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>
<dt>arrow.min.intron.length</dt>
<dd><p><code><a href="https://rdrr.io/r/base/integer.html" class="external-link">integer()</a></code> the minimum required width of an
intron for a strand arrow to be drawn. This can be useful to remove strand
arrows on short introns that overlap adjacent exons.</p></dd>
<dt>show.legend</dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>
<dt>inherit.aes</dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>geom_intron()</code>  requires the following <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>; <code>xstart</code>, <code>xend</code> and <code>y</code>
(e.g. transcript name). The <code>strand</code> option (one of "+" or "-") adjusts the
arrow direction to match the direction of transcription. the
<code>arrow.min.intron.length</code> parameter can be useful to remove strand arrows
overlapping exons, which can be a problem if plotted introns include those
that are relatively short.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># to illustrate the package's functionality</span></span>
<span class="r-in"><span class="co"># ggtranscript includes example transcript annotation</span></span>
<span class="r-in"><span class="va">pknox1_annotation</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 112 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames    start      end strand type        gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      gene        PKNOX1    <span style="color: #BB0000;">NA</span>             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      transcript  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon        PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon        PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon        PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon        PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      CDS         PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>070 +      start_codon PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> 21       43<span style="text-decoration: underline;">016</span>908 43<span style="text-decoration: underline;">017</span>007 +      exon        PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> 21       43<span style="text-decoration: underline;">016</span>908 43<span style="text-decoration: underline;">017</span>007 +      CDS         PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 102 more rows, and 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># extract exons</span></span>
<span class="r-in"><span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">pknox1_exons</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 59 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames    start      end strand type  gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> 21       43<span style="text-decoration: underline;">016</span>908 43<span style="text-decoration: underline;">017</span>007 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> 21       43<span style="text-decoration: underline;">018</span>133 43<span style="text-decoration: underline;">018</span>230 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> 21       43<span style="text-decoration: underline;">021</span>303 43<span style="text-decoration: underline;">021</span>431 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> 21       43<span style="text-decoration: underline;">024</span>871 43<span style="text-decoration: underline;">024</span>947 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> 21       43<span style="text-decoration: underline;">028</span>702 43<span style="text-decoration: underline;">028</span>874 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> 21       43<span style="text-decoration: underline;">029</span>890 43<span style="text-decoration: underline;">033</span>931 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 49 more rows, and 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># to_intron() is a helper function included in ggtranscript</span></span>
<span class="r-in"><span class="co"># which is useful for converting exon co-ordinates to introns</span></span>
<span class="r-in"><span class="va">pknox1_introns</span> <span class="op">&lt;-</span> <span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span>group_var <span class="op">=</span> <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">pknox1_introns</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 48 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames strand type   gene_name transcript_name transcript_biotype     start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> 21       +      intron PKNOX1    PKNOX1-203      protein_coding        4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> 21       +      intron PKNOX1    PKNOX1-202      retained_intron       4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> 21       +      intron PKNOX1    PKNOX1-201      protein_coding        4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> 21       +      intron PKNOX1    PKNOX1-206      retained_intron       4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> 21       +      intron PKNOX1    PKNOX1-204      retained_intron       4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> 21       +      intron PKNOX1    PKNOX1-203      protein_coding        4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> 21       +      intron PKNOX1    PKNOX1-201      protein_coding        4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> 21       +      intron PKNOX1    PKNOX1-206      retained_intron       4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> 21       +      intron PKNOX1    PKNOX1-209      nonsense_mediated_de… 4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> 21       +      intron PKNOX1    PKNOX1-202      retained_intron       4.30<span style="color: #949494;">e</span>7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 38 more rows, and 1 more variable: end &lt;int&gt;</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">base</span> <span class="op">&lt;-</span> <span class="va">pknox1_introns</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">        <span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># by default, geom_intron() assumes introns originate from the "+" strand</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_intron-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># however this can be modified using the strand option</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_intron-2.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># strand can also be set as an aes()</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_intron-3.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># as a ggplot2 extension, ggtranscript geoms inherit the</span></span>
<span class="r-in"><span class="co"># the functionality from the parameters and aesthetics in ggplot2</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span></span>
<span class="r-in">  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">transcript_name</span><span class="op">)</span>,</span>
<span class="r-in">  size <span class="op">=</span> <span class="fl">1</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_intron-4.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># together, geom_range() and geom_range() are designed to visualize</span></span>
<span class="r-in"><span class="co"># the core components of transcript annotation</span></span>
<span class="r-in"><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">  <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">  <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">  <span class="fu">geom_intron</span><span class="op">(</span></span>
<span class="r-in">    data <span class="op">=</span> <span class="va">pknox1_introns</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_intron-5.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># for short introns, sometimes strand arrows will overlap exons</span></span>
<span class="r-in"><span class="co"># to avoid this, users can set the arrow.min.intron.length parameter</span></span>
<span class="r-in"><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">  <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">  <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">  <span class="fu">geom_intron</span><span class="op">(</span></span>
<span class="r-in">    data <span class="op">=</span> <span class="va">pknox1_introns</span>,</span>
<span class="r-in">    arrow.min.intron.length <span class="op">=</span> <span class="fl">3500</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_intron-6.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

