<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot intron lines with strand arrows — geom_intron • ggtranscript</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot intron lines with strand arrows — geom_intron"><meta name="description" content="geom_intron() draws horizontal lines with central arrows that are designed
to represent introns. In combination with geom_range()/geom_half_range(),
these geoms form the core components for visualizing transcript structures."><meta property="og:description" content="geom_intron() draws horizontal lines with central arrows that are designed
to represent introns. In combination with geom_range()/geom_half_range(),
these geoms form the core components for visualizing transcript structures."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/ggtranscript.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Plot intron lines with strand arrows</h1>
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/master/R/geom_intron.R" class="external-link"><code>R/geom_intron.R</code></a></small>
      <div class="d-none name"><code>geom_intron.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>geom_intron()</code> draws horizontal lines with central arrows that are designed
to represent introns. In combination with <code><a href="geom_range.html">geom_range()</a></code>/<code><a href="geom_range.html">geom_half_range()</a></code>,
these geoms form the core components for visualizing transcript structures.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">geom_intron</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  arrow <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></span><span class="op">(</span>ends <span class="op">=</span> <span class="st">"last"</span>, length <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="st">"inches"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  arrow.fill <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  lineend <span class="op">=</span> <span class="st">"butt"</span>,</span>
<span>  linejoin <span class="op">=</span> <span class="st">"round"</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  arrow.min.intron.length <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-mapping">mapping<a class="anchor" aria-label="anchor" href="#arg-mapping"></a></dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p></dd>


<dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="https://ggplot2.tidyverse.org/reference/fortify.html" class="external-link">fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></dd>


<dt id="arg-stat">stat<a class="anchor" aria-label="anchor" href="#arg-stat"></a></dt>
<dd><p>The statistical transformation to use on the data for this layer.
When using a <code>geom_*()</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:</p><ul><li><p>A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.</p></li>
<li><p>A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">stat_count()</a></code>,
give the stat as <code>"count"</code>.</p></li>
<li><p>For more information and other ways to specify the stat, see the
<a href="https://ggplot2.tidyverse.org/reference/layer_stats.html" class="external-link">layer stat</a> documentation.</p></li>
</ul></dd>


<dt id="arg-position">position<a class="anchor" aria-label="anchor" href="#arg-position"></a></dt>
<dd><p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:</p><ul><li><p>The result of calling a position function, such as <code><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter()</a></code>.
This method allows for passing extra arguments to the position.</p></li>
<li><p>A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter()</a></code>, give the position as <code>"jitter"</code>.</p></li>
<li><p>For more information and other ways to specify the position, see the
<a href="https://ggplot2.tidyverse.org/reference/layer_positions.html" class="external-link">layer position</a> documentation.</p></li>
</ul></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code>'s <code>params</code> argument. These
arguments broadly fall into one of 4 categories below. Notably, further
arguments to the <code>position</code> argument, or aesthetics that are required
can <em>not</em> be passed through <code>...</code>. Unknown arguments that are not part
of the 4 categories below are ignored.</p><ul><li><p>Static aesthetics that are not mapped to a scale, but are at a fixed
value and apply to the layer as a whole. For example, <code>colour = "red"</code>
or <code>linewidth = 3</code>. The geom's documentation has an <strong>Aesthetics</strong>
section that lists the available options. The 'required' aesthetics
cannot be passed on to the <code>params</code>. Please note that while passing
unmapped aesthetics as vectors is technically possible, the order and
required length is not guaranteed to be parallel to the input data.</p></li>
<li><p>When constructing a layer using
a <code>stat_*()</code> function, the <code>...</code> argument can be used to pass on
parameters to the <code>geom</code> part of the layer. An example of this is
<code>stat_density(geom = "area", outline.type = "both")</code>. The geom's
documentation lists which parameters it can accept.</p></li>
<li><p>Inversely, when constructing a layer using a
<code>geom_*()</code> function, the <code>...</code> argument can be used to pass on parameters
to the <code>stat</code> part of the layer. An example of this is
<code>geom_area(stat = "density", adjust = 0.5)</code>. The stat's documentation
lists which parameters it can accept.</p></li>
<li><p>The <code>key_glyph</code> argument of <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code> may also be passed on through
<code>...</code>. This can be one of the functions described as
<a href="https://ggplot2.tidyverse.org/reference/draw_key.html" class="external-link">key glyphs</a>, to change the display of the layer in the legend.</p></li>
</ul></dd>


<dt id="arg-arrow">arrow<a class="anchor" aria-label="anchor" href="#arg-arrow"></a></dt>
<dd><p>specification for arrow heads, as created by <code><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">grid::arrow()</a></code>.</p></dd>


<dt id="arg-arrow-fill">arrow.fill<a class="anchor" aria-label="anchor" href="#arg-arrow-fill"></a></dt>
<dd><p>fill colour to use for the arrow head (if closed). <code>NULL</code>
means use <code>colour</code> aesthetic.</p></dd>


<dt id="arg-lineend">lineend<a class="anchor" aria-label="anchor" href="#arg-lineend"></a></dt>
<dd><p>Line end style (round, butt, square).</p></dd>


<dt id="arg-linejoin">linejoin<a class="anchor" aria-label="anchor" href="#arg-linejoin"></a></dt>
<dd><p>Line join style (round, mitre, bevel).</p></dd>


<dt id="arg-na-rm">na.rm<a class="anchor" aria-label="anchor" href="#arg-na-rm"></a></dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>


<dt id="arg-arrow-min-intron-length">arrow.min.intron.length<a class="anchor" aria-label="anchor" href="#arg-arrow-min-intron-length"></a></dt>
<dd><p><code><a href="https://rdrr.io/r/base/integer.html" class="external-link">integer()</a></code> the minimum required width of an
intron for a strand arrow to be drawn. This can be useful to remove strand
arrows on short introns that overlap adjacent exons.</p></dd>


<dt id="arg-show-legend">show.legend<a class="anchor" aria-label="anchor" href="#arg-show-legend"></a></dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>


<dt id="arg-inherit-aes">inherit.aes<a class="anchor" aria-label="anchor" href="#arg-inherit-aes"></a></dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>the return value of a <code>geom_*</code> function is not intended to be
directly handled by users. Therefore, <code>geom_*</code> functions should never be
executed in isolation, rather used in combination with a
<code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2::ggplot()</a></code> call.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>geom_intron()</code> requires the following <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>; <code>xstart</code>, <code>xend</code> and <code>y</code>
(e.g. transcript name). If users do not have intron co-ordinates, these can
be generated from the corresponding exons using <code><a href="to_intron.html">to_intron()</a></code>. The <code>strand</code>
option (one of "+" or "-") adjusts the arrow direction to match the direction
of transcription. The <code>arrow.min.intron.length</code> parameter can be useful to
remove strand arrows that overlap exons, which can be a problem if plotted
introns include those that are relatively short.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to illustrate the package's functionality</span></span></span>
<span class="r-in"><span><span class="co"># ggtranscript includes example transcript annotation</span></span></span>
<span class="r-in"><span><span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames    start      end strand type       gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      gene       PKNOX1    <span style="color: #BB0000;">NA</span>             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      transcript PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># extract exons</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames    start      end strand type  gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       43<span style="text-decoration: underline;">016</span>908 43<span style="text-decoration: underline;">017</span>007 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       43<span style="text-decoration: underline;">018</span>133 43<span style="text-decoration: underline;">018</span>230 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to_intron() is a helper function included in ggtranscript</span></span></span>
<span class="r-in"><span><span class="co"># which is useful for converting exon co-ordinates to introns</span></span></span>
<span class="r-in"><span><span class="va">pknox1_introns</span> <span class="op">&lt;-</span> <span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span>group_var <span class="op">=</span> <span class="st">"transcript_name"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pknox1_introns</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames strand type   gene_name transcript_name transcript_biotype    start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       +      intron PKNOX1    PKNOX1-203      protein_coding     42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       +      intron PKNOX1    PKNOX1-202      retained_intron    42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       +      intron PKNOX1    PKNOX1-201      protein_coding     42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       +      intron PKNOX1    PKNOX1-206      retained_intron    42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       +      intron PKNOX1    PKNOX1-204      retained_intron    43<span style="text-decoration: underline;">004</span>432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       +      intron PKNOX1    PKNOX1-203      protein_coding     43<span style="text-decoration: underline;">004</span>432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: end &lt;int&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">&lt;-</span> <span class="va">pknox1_introns</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        xstart <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>        xend <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        y <span class="op">=</span> <span class="va">transcript_name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># by default, geom_intron() assumes introns originate from the "+" strand</span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_intron-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># however this can be modified using the strand option</span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_intron-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># strand can also be set as an aes()</span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_intron-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># as a ggplot2 extension, ggtranscript geoms inherit the</span></span></span>
<span class="r-in"><span><span class="co"># the functionality from the parameters and aesthetics in ggplot2</span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_intron</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">transcript_name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    linewidth <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_intron-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># together, geom_range() and geom_intron() are designed to visualize</span></span></span>
<span class="r-in"><span><span class="co"># the core components of transcript annotation</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        xstart <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>        xend <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        y <span class="op">=</span> <span class="va">transcript_name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu">geom_intron</span><span class="op">(</span></span></span>
<span class="r-in"><span>        data <span class="op">=</span> <span class="va">pknox1_introns</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_intron-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># for short introns, sometimes strand arrows will overlap exons</span></span></span>
<span class="r-in"><span><span class="co"># to avoid this, users can set the arrow.min.intron.length parameter</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        xstart <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>        xend <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        y <span class="op">=</span> <span class="va">transcript_name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu">geom_intron</span><span class="op">(</span></span></span>
<span class="r-in"><span>        data <span class="op">=</span> <span class="va">pknox1_introns</span>,</span></span>
<span class="r-in"><span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">3500</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_intron-6.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

