<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="geom_junction_label_repel() labels junction curves at their midpoint using
ggrepel::geom_label_repel(). This can be useful to label and compare
junctions (plotted using geom_junction()) with metrics of their usage (e.g.
read counts or percent-spliced-in)."><title>Label junction curves — geom_junction_label_repel • ggtranscript</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Label junction curves — geom_junction_label_repel"><meta property="og:description" content="geom_junction_label_repel() labels junction curves at their midpoint using
ggrepel::geom_label_repel(). This can be useful to label and compare
junctions (plotted using geom_junction()) with metrics of their usage (e.g.
read counts or percent-spliced-in)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/ggtranscript.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Label junction curves</h1>
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/geom_junction_label_repel.R" class="external-link"><code>R/geom_junction_label_repel.R</code></a></small>
      <div class="d-none name"><code>geom_junction_label_repel.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>geom_junction_label_repel()</code> labels junction curves at their midpoint using
<code><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">ggrepel::geom_label_repel()</a></code>. This can be useful to label and compare
junctions (plotted using <code><a href="geom_junction.html">geom_junction()</a></code>) with metrics of their usage (e.g.
read counts or percent-spliced-in).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">geom_junction_label_repel</span><span class="op">(</span>
  mapping <span class="op">=</span> <span class="cn">NULL</span>,
  data <span class="op">=</span> <span class="cn">NULL</span>,
  stat <span class="op">=</span> <span class="st">"identity"</span>,
  position <span class="op">=</span> <span class="st">"identity"</span>,
  parse <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">...</span>,
  junction.orientation <span class="op">=</span> <span class="st">"alternating"</span>,
  junction.y.max <span class="op">=</span> <span class="fl">1</span>,
  angle <span class="op">=</span> <span class="fl">90</span>,
  ncp <span class="op">=</span> <span class="fl">15</span>,
  box.padding <span class="op">=</span> <span class="fl">0.25</span>,
  label.padding <span class="op">=</span> <span class="fl">0.25</span>,
  point.padding <span class="op">=</span> <span class="fl">1e-06</span>,
  label.r <span class="op">=</span> <span class="fl">0.15</span>,
  label.size <span class="op">=</span> <span class="fl">0.25</span>,
  min.segment.length <span class="op">=</span> <span class="fl">0</span>,
  arrow <span class="op">=</span> <span class="cn">NULL</span>,
  force <span class="op">=</span> <span class="fl">1</span>,
  force_pull <span class="op">=</span> <span class="fl">1</span>,
  max.time <span class="op">=</span> <span class="fl">0.5</span>,
  max.iter <span class="op">=</span> <span class="fl">10000</span>,
  max.overlaps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"ggrepel.max.overlaps"</span>, default <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
  nudge_x <span class="op">=</span> <span class="fl">0</span>,
  nudge_y <span class="op">=</span> <span class="fl">0</span>,
  xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="cn">NA</span><span class="op">)</span>,
  ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="cn">NA</span><span class="op">)</span>,
  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,
  show.legend <span class="op">=</span> <span class="cn">NA</span>,
  direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"both"</span>, <span class="st">"y"</span>, <span class="st">"x"</span><span class="op">)</span>,
  seed <span class="op">=</span> <span class="cn">NA</span>,
  verbose <span class="op">=</span> <span class="cn">FALSE</span>,
  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mapping</dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></code> or
<code><a href="https://ggplot2.tidyverse.org/reference/aes_.html" class="external-link">aes_</a></code>. If specified and <code>inherit.aes = TRUE</code> (the
default), is combined with the default mapping at the top level of the
plot. You only need to supply <code>mapping</code> if there isn't a mapping
defined for the plot.</p></dd>
<dt>data</dt>
<dd><p>A data frame. If specified, overrides the default data frame
defined at the top level of the plot.</p></dd>
<dt>stat</dt>
<dd><p>The statistical transformation to use on the data for this
layer, as a string.</p></dd>
<dt>position</dt>
<dd><p>Position adjustment, either as a string, or the result of
a call to a position adjustment function.</p></dd>
<dt>parse</dt>
<dd><p>If TRUE, the labels will be parsed into expressions and
displayed as described in ?plotmath</p></dd>
<dt>...</dt>
<dd><p>other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer</a></code>. There are
  three types of arguments you can use here:</p>
<p></p><ul><li><p>Aesthetics: to set an aesthetic to a fixed value, like
       <code>colour = "red"</code> or <code>size = 3</code>.</p></li>
<li><p>Other arguments to the layer, for example you override the
      default <code>stat</code> associated with the layer.</p></li>
<li><p>Other arguments passed on to the stat.</p></li>
</ul></dd>
<dt>junction.orientation</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> one of "alternating", "top" or
"bottom", specifying where the junctions will be plotted with respect to
each transcript (<code>y</code>).</p></dd>
<dt>junction.y.max</dt>
<dd><p><code><a href="https://rdrr.io/r/base/double.html" class="external-link">double()</a></code> the max y-value of each junction curve. It
can be useful to adjust this parameter when junction curves overlap with
one another/other transcripts or extend beyond the plot margins.</p></dd>
<dt>angle</dt>
<dd><p>A numeric value between 0 and 180,
    giving an amount to skew the control
    points of the curve.  Values less than 90 skew the curve towards
    the start point and values greater than 90 skew the curve
    towards the end point.</p></dd>
<dt>ncp</dt>
<dd><p>The number of control points used to draw the curve.
    More control points creates a smoother curve.</p></dd>
<dt>box.padding</dt>
<dd><p>Amount of padding around bounding box, as unit or number.
Defaults to 0.25. (Default unit is lines, but other units can be specified
by passing <code>unit(x, "units")</code>).</p></dd>
<dt>label.padding</dt>
<dd><p>Amount of padding around label, as unit or number.
Defaults to 0.25. (Default unit is lines, but other units can be specified
by passing <code>unit(x, "units")</code>).</p></dd>
<dt>point.padding</dt>
<dd><p>Amount of padding around labeled point, as unit or
number. Defaults to 0. (Default unit is lines, but other units can be
specified by passing <code>unit(x, "units")</code>).</p></dd>
<dt>label.r</dt>
<dd><p>Radius of rounded corners, as unit or number. Defaults
to 0.15. (Default unit is lines, but other units can be specified by
passing <code>unit(x, "units")</code>).</p></dd>
<dt>label.size</dt>
<dd><p>Size of label border, in mm.</p></dd>
<dt>min.segment.length</dt>
<dd><p>Skip drawing segments shorter than this, as unit or
number. Defaults to 0.5. (Default unit is lines, but other units can be
specified by passing <code>unit(x, "units")</code>).</p></dd>
<dt>arrow</dt>
<dd><p>specification for arrow heads, as created by <code><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></code></p></dd>
<dt>force</dt>
<dd><p>Force of repulsion between overlapping text labels. Defaults
to 1.</p></dd>
<dt>force_pull</dt>
<dd><p>Force of attraction between a text label and its
corresponding data point. Defaults to 1.</p></dd>
<dt>max.time</dt>
<dd><p>Maximum number of seconds to try to resolve overlaps.
Defaults to 0.5.</p></dd>
<dt>max.iter</dt>
<dd><p>Maximum number of iterations to try to resolve overlaps.
Defaults to 10000.</p></dd>
<dt>max.overlaps</dt>
<dd><p>Exclude text labels that overlap too many things.
Defaults to 10.</p></dd>
<dt>nudge_x</dt>
<dd><p>Horizontal and vertical adjustments to nudge the
starting position of each text label. The units for <code>nudge_x</code> and
<code>nudge_y</code> are the same as for the data units on the x-axis and y-axis.</p></dd>
<dt>nudge_y</dt>
<dd><p>Horizontal and vertical adjustments to nudge the
starting position of each text label. The units for <code>nudge_x</code> and
<code>nudge_y</code> are the same as for the data units on the x-axis and y-axis.</p></dd>
<dt>xlim</dt>
<dd><p>Limits for the x and y axes. Text labels will be constrained
to these limits. By default, text labels are constrained to the entire plot
area.</p></dd>
<dt>ylim</dt>
<dd><p>Limits for the x and y axes. Text labels will be constrained
to these limits. By default, text labels are constrained to the entire plot
area.</p></dd>
<dt>na.rm</dt>
<dd><p>If <code>FALSE</code> (the default), removes missing values with
a warning.  If <code>TRUE</code> silently removes missing values.</p></dd>
<dt>show.legend</dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.</p></dd>
<dt>direction</dt>
<dd><p>"both", "x", or "y" -- direction in which to adjust position of labels</p></dd>
<dt>seed</dt>
<dd><p>Random seed passed to <code><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></code>. Defaults to
<code>NA</code>, which means that <code>set.seed</code> will not be called.</p></dd>
<dt>verbose</dt>
<dd><p>If <code>TRUE</code>, some diagnostics of the repel algorithm are printed</p></dd>
<dt>inherit.aes</dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders</a></code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>the return value of a <code>geom_*</code> function is not intended to be
directly handled by users. Therefore, <code>geom_*</code> functions should never be
executed in isolation, rather used in combination with a
<code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2::ggplot()</a></code> call.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>geom_junction_label_repel()</code> requires the following <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>; <code>xstart</code>,
<code>xend</code>, <code>y</code> (e.g. transcript name) and <code>label</code>. Under the hood,
<code>geom_junction_label_repel()</code> generates the same junction curves as
<code><a href="geom_junction.html">geom_junction()</a></code> to obtain curve midpoints for labeling. Therefore, it is
important that users use the same input data and parameters that alter
junction curves (namely <code>junction.orientation</code>, <code>junction.y.max</code>, <code>angle</code>,
<code>ncp</code>) for <code>geom_junction_label_repel()</code> that they have used for
<code><a href="geom_junction.html">geom_junction()</a></code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># to illustrate the package's functionality</span></span>
<span class="r-in"><span class="co"># ggtranscript includes example transcript annotation</span></span>
<span class="r-in"><span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      gene  SOD1      <span style="color: #BB0000;">NA</span>              <span style="color: #BB0000;">NA</span>              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      tran… SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      CDS   SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      star… SOD1      SOD1-202        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># as well as a set of example (unannotated) junctions</span></span>
<span class="r-in"><span class="co"># originating from GTEx and downloaded via the Bioconductor package snapcount</span></span>
<span class="r-in"><span class="va">sod1_junctions</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 5 × 5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames    start      end strand mean_count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> chr21    31<span style="text-decoration: underline;">659</span>787 31<span style="text-decoration: underline;">666</span>448 +           0.463</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">660</span>554 +           0.831</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">663</span>794 +           0.316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">667</span>257 +           4.35 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> chr21    31<span style="text-decoration: underline;">660</span>351 31<span style="text-decoration: underline;">663</span>789 +           0.324</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># extract exons</span></span>
<span class="r-in"><span class="va">sod1_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span class="r-in">    <span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span>,</span>
<span class="r-in">    <span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"><span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 5 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-201        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-201        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-201        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-201        protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-201        protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># add transcript_name to junctions for plotting</span></span>
<span class="r-in"><span class="va">sod1_junctions</span> <span class="op">&lt;-</span> <span class="va">sod1_junctions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>transcript_name <span class="op">=</span> <span class="st">"SOD1-201"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># geom_junction_label_repel() can be used to label junctions</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># this can be useful to label junctions with their counts</span></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu"><a href="geom_junction.html">geom_junction</a></span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">sod1_junctions</span>,</span>
<span class="r-in">        junction.y.max <span class="op">=</span> <span class="fl">0.5</span></span>
<span class="r-in">    <span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">    <span class="fu">geom_junction_label_repel</span><span class="op">(</span></span>
<span class="r-in">        data <span class="op">=</span> <span class="va">sod1_junctions</span>,</span>
<span class="r-in">        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">mean_count</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span>
<span class="r-in">        junction.y.max <span class="op">=</span> <span class="fl">0.5</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction_label_repel-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

