<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Improve transcript structure visualization by shortening gaps — shorten_gaps • ggtranscript</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Improve transcript structure visualization by shortening gaps — shorten_gaps"><meta name="description" content="For a given set of exons and introns, shorten_gaps() reduces the width of
gaps (regions that do not overlap any exons) to a user-inputted
target_gap_width. This can be useful when visualizing transcripts that have
long introns, to hone in on the regions of interest (i.e. exons) and better
compare between transcript structures."><meta property="og:description" content="For a given set of exons and introns, shorten_gaps() reduces the width of
gaps (regions that do not overlap any exons) to a user-inputted
target_gap_width. This can be useful when visualizing transcripts that have
long introns, to hone in on the regions of interest (i.e. exons) and better
compare between transcript structures."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/ggtranscript.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Improve transcript structure visualization by shortening gaps</h1>
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/master/R/shorten_gaps.R" class="external-link"><code>R/shorten_gaps.R</code></a></small>
      <div class="d-none name"><code>shorten_gaps.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>For a given set of exons and introns, <code>shorten_gaps()</code> reduces the width of
gaps (regions that do not overlap any <code>exons</code>) to a user-inputted
<code>target_gap_width</code>. This can be useful when visualizing transcripts that have
long introns, to hone in on the regions of interest (i.e. exons) and better
compare between transcript structures.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">shorten_gaps</span><span class="op">(</span><span class="va">exons</span>, <span class="va">introns</span>, group_var <span class="op">=</span> <span class="cn">NULL</span>, target_gap_width <span class="op">=</span> <span class="fl">100L</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-exons">exons<a class="anchor" aria-label="anchor" href="#arg-exons"></a></dt>
<dd><p><code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> contains exons which can originate from multiple
transcripts differentiated by <code>group_var</code>.</p></dd>


<dt id="arg-introns">introns<a class="anchor" aria-label="anchor" href="#arg-introns"></a></dt>
<dd><p><code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> the intron co-ordinates corresponding to the
input <code>exons</code>. This can be created by applying <code><a href="to_intron.html">to_intron()</a></code> to the
<code>exons</code>. If introns originate from multiple transcripts, they must be
differentiated using <code>group_var</code>. If a user is not using <code><a href="to_intron.html">to_intron()</a></code>,
they must make sure intron start/ends are defined precisely as the adjacent
exon boundaries (rather than exon end + 1 and exon start - 1).</p></dd>


<dt id="arg-group-var">group_var<a class="anchor" aria-label="anchor" href="#arg-group-var"></a></dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> if input data originates from more than 1
transcript, <code>group_var</code> must specify the column that differentiates
transcripts (e.g. "transcript_id").</p></dd>


<dt id="arg-target-gap-width">target_gap_width<a class="anchor" aria-label="anchor" href="#arg-target-gap-width"></a></dt>
<dd><p><code><a href="https://rdrr.io/r/base/integer.html" class="external-link">integer()</a></code> the width in base pairs to shorten the
gaps to.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> contains the re-scaled co-ordinates of <code>introns</code> and
<code>exons</code> of each input transcript with shortened gaps.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>After <code>shorten_gaps()</code> reduces the size of gaps, it will re-scale <code>exons</code> and
<code>introns</code> to preserve exon alignment. This process will only reduce the width
of input <code>introns</code>, never <code>exons</code>. Importantly, the outputted re-scaled
co-ordinates should only be used for visualization as they will not match the
original genomic coordinates.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to illustrate the package's functionality</span></span></span>
<span class="r-in"><span><span class="co"># ggtranscript includes example transcript annotation</span></span></span>
<span class="r-in"><span><span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames    start      end strand type       gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      gene       PKNOX1    <span style="color: #BB0000;">NA</span>             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      transcript PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon       PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># extract exons</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames    start      end strand type  gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       43<span style="text-decoration: underline;">016</span>908 43<span style="text-decoration: underline;">017</span>007 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       43<span style="text-decoration: underline;">018</span>133 43<span style="text-decoration: underline;">018</span>230 +      exon  PKNOX1    PKNOX1-203     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to_intron() is a helper function included in ggtranscript</span></span></span>
<span class="r-in"><span><span class="co"># which is useful for converting exon co-ordinates to introns</span></span></span>
<span class="r-in"><span><span class="va">pknox1_introns</span> <span class="op">&lt;-</span> <span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span>group_var <span class="op">=</span> <span class="st">"transcript_name"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pknox1_introns</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames strand type   gene_name transcript_name transcript_biotype    start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21       +      intron PKNOX1    PKNOX1-203      protein_coding     42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21       +      intron PKNOX1    PKNOX1-202      retained_intron    42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21       +      intron PKNOX1    PKNOX1-201      protein_coding     42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21       +      intron PKNOX1    PKNOX1-206      retained_intron    42<span style="text-decoration: underline;">974</span>664</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21       +      intron PKNOX1    PKNOX1-204      retained_intron    43<span style="text-decoration: underline;">004</span>432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21       +      intron PKNOX1    PKNOX1-203      protein_coding     43<span style="text-decoration: underline;">004</span>432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: end &lt;int&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># for transcripts with long introns, the exons of interest</span></span></span>
<span class="r-in"><span><span class="co"># can be difficult to visualize clearly when using the default scale</span></span></span>
<span class="r-in"><span><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        xstart <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>        xend <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        y <span class="op">=</span> <span class="va">transcript_name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        data <span class="op">=</span> <span class="va">pknox1_introns</span>,</span></span>
<span class="r-in"><span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">3500</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-plt img"><img src="shorten_gaps-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># in such cases it can be useful to rescale the exons and introns</span></span></span>
<span class="r-in"><span><span class="co"># using shorten_gaps() which shortens regions that do not overlap an exon</span></span></span>
<span class="r-in"><span><span class="va">pknox1_rescaled</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>    <span class="fu">shorten_gaps</span><span class="op">(</span><span class="va">pknox1_exons</span>, <span class="va">pknox1_introns</span>, group_var <span class="op">=</span> <span class="st">"transcript_name"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqnames start   end strand type  gene_name transcript_name transcript_biotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 21          54   156 +      exon  PKNOX1    PKNOX1-201      protein_coding    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 21         156   694 +      intr… PKNOX1    PKNOX1-201      protein_coding    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> 21         694   800 +      exon  PKNOX1    PKNOX1-201      protein_coding    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> 21         800   901 +      intr… PKNOX1    PKNOX1-201      protein_coding    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> 21         901  <span style="text-decoration: underline;">1</span>028 +      exon  PKNOX1    PKNOX1-201      protein_coding    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> 21        <span style="text-decoration: underline;">1</span>028  <span style="text-decoration: underline;">1</span>223 +      intr… PKNOX1    PKNOX1-201      protein_coding    </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># this allows us to visualize differences in exonic structure more clearly</span></span></span>
<span class="r-in"><span><span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        xstart <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>        xend <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        y <span class="op">=</span> <span class="va">transcript_name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        data <span class="op">=</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"intron"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-plt img"><img src="shorten_gaps-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># shorten_gaps() can be used in combination with to_diff()</span></span></span>
<span class="r-in"><span><span class="co"># to further highlight differences in exon structure</span></span></span>
<span class="r-in"><span><span class="co"># here, all other transcripts are compared to the MANE-select transcript</span></span></span>
<span class="r-in"><span><span class="va">pknox1_rescaled_diffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="to_diff.html">to_diff</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    exons <span class="op">=</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span>, <span class="va">transcript_name</span> <span class="op">!=</span> <span class="st">"PKNOX1-201"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    ref_exons <span class="op">=</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span>, <span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"PKNOX1-201"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        xstart <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>        xend <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        y <span class="op">=</span> <span class="va">transcript_name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        data <span class="op">=</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"intron"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span></span></span>
<span class="r-in"><span>    <span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        data <span class="op">=</span> <span class="va">pknox1_rescaled_diffs</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">diff_type</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        alpha <span class="op">=</span> <span class="fl">0.2</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-plt img"><img src="shorten_gaps-3.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

