<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started • ggtranscript</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/ggtranscript.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started</h1>
                        <h4 data-toc-skip class="author">David
Zhang</h4>
            <address class="author_afil">
      UCL<br><a class="author_email" href="mailto:#"></a><a href="mailto:dyzhang32@gmail.com" class="email">dyzhang32@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/master/vignettes/ggtranscript.Rmd" class="external-link"><code>vignettes/ggtranscript.Rmd</code></a></small>
      <div class="d-none name"><code>ggtranscript.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dzhang32/ggtranscript" class="external-link">ggtranscript</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     combine, intersect, setdiff, union</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     first, rename</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'IRanges'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     collapse, desc, slice</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'GenomicRanges'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:magrittr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     subtract</span></span></code></pre></div>
<p><code>ggtranscript</code> is designed to make it easy to visualize
transcript structure and annotation using <code>ggplot2</code>.</p>
<p>As the intended users are those who work with genetic and/or
transcriptomic data in <code>R</code>, this tutorial assumes a basic
understanding of transcript annotation and familiarity with
<code>ggplot2</code>.</p>
<p><br></p>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<div class="section level3">
<h3 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>In order to showcase the package’s functionality,
<code>ggtranscript</code> includes example transcript annotation for the
genes <em>SOD1</em> and <em>PKNOX1</em>, as well as a set of unannotated
junctions associated with <em>SOD1</em>. These specific genes are
unimportant, chosen arbitrarily for illustration. However, it worth
noting that the input data for <code>ggtranscript</code>, as a
<code>ggplot2</code> extension, is required be a <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/data.frame" class="external-link"><code>data.frame</code></a>
or <a href="https://tibble.tidyverse.org" class="external-link"><code>tibble</code></a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;   seqnames    start      end strand type        gene_name transcript_name</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">668</span>931 +      gene        SOD1      <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">668</span>931 +      transcript  SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">659</span>784 +      exon        SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       31<span style="text-decoration: underline;">659</span>770 31<span style="text-decoration: underline;">659</span>784 +      CDS         SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       31<span style="text-decoration: underline;">659</span>770 31<span style="text-decoration: underline;">659</span>772 +      start_codon SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       31<span style="text-decoration: underline;">663</span>790 31<span style="text-decoration: underline;">663</span>886 +      exon        SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;   seqnames    start      end strand type       gene_name transcript_name</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      gene       PKNOX1    <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       42<span style="text-decoration: underline;">974</span>510 43<span style="text-decoration: underline;">033</span>931 +      transcript PKNOX1    PKNOX1-203     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       42<span style="text-decoration: underline;">974</span>510 42<span style="text-decoration: underline;">974</span>664 +      exon       PKNOX1    PKNOX1-203     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       43<span style="text-decoration: underline;">004</span>326 43<span style="text-decoration: underline;">004</span>432 +      exon       PKNOX1    PKNOX1-203     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       43<span style="text-decoration: underline;">007</span>491 43<span style="text-decoration: underline;">007</span>618 +      exon       PKNOX1    PKNOX1-203     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       43<span style="text-decoration: underline;">013</span>068 43<span style="text-decoration: underline;">013</span>238 +      exon       PKNOX1    PKNOX1-203     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="va">sod1_junctions</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 5</span></span></span>
<span><span class="co">#&gt;   seqnames    start      end strand mean_count</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr21    31<span style="text-decoration: underline;">659</span>787 31<span style="text-decoration: underline;">666</span>448 +           0.463</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">660</span>554 +           0.831</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">663</span>794 +           0.316</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">667</span>257 +           4.35 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chr21    31<span style="text-decoration: underline;">660</span>351 31<span style="text-decoration: underline;">663</span>789 +           0.324</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="importing-data-from-a-gtf">Importing data from a gtf<a class="anchor" aria-label="anchor" href="#importing-data-from-a-gtf"></a>
</h3>
<p>You may be asking, what if I have a <code>gtf</code> file or a
<code>GRanges</code> object? The below demonstrates how to wrangle a
<code>gtf</code> into the required format for <code>ggtranscript</code>
and extract the relevant annotation for a particular gene of
interest.</p>
<p>For the purposes of the vignette, here we download a <code>gtf</code>
(Ensembl version 105), then load the <code>gtf</code> into
<code>R</code> using <code>rtracklayer::import()</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># download ens 105 gtf into a temporary directory</span></span>
<span><span class="va">gtf_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"Homo_sapiens.GRCh38.105.chr.gtf.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>        <span class="st">"http://ftp.ensembl.org/pub/release-105/gtf/homo_sapiens/"</span>,</span>
<span>        <span class="st">"Homo_sapiens.GRCh38.105.chr.gtf.gz"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    destfile <span class="op">=</span> <span class="va">gtf_path</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">gtf_path</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GRanges"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "GenomicRanges"</span></span></code></pre></div>
<p>To note, the loaded <code>gtf</code> is a <code>GRanges</code> class
object. The input data for <code>ggtranscript</code>, as a
<code>ggplot2</code> extension, is required be a <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/data.frame" class="external-link"><code>data.frame</code></a>
or <a href="https://tibble.tidyverse.org" class="external-link"><code>tibble</code></a>. We
can convert a <code>GRanges</code> to a <code>data.frame</code> using
<code>as.data.frame</code> or a <code>tibble</code> via
<code><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">dplyr::as_tibble()</a></code>. Either is fine with respect to
<code>ggtranscript</code>, however we prefer <code>tibble</code>s over
<code>data.frame</code>s for several <a href="https://r4ds.had.co.nz/tibbles.html#tibbles-vs.-data.frame" class="external-link">reasons</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="va">gtf</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "tbl_df"     "tbl"        "data.frame"</span></span></code></pre></div>
<p>Now that the <code>gtf</code> is a <code>tibble</code> (or
<code>data.frame</code> object), we can <code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">dplyr::filter()</a></code>
rows and <code><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">dplyr::select()</a></code> columns to keep the annotation
columns required for any specific gene of interest. Here, we illustrate
how you would obtain the annotation for the gene <em>SOD1</em>, ready
for plotting with <code>ggtranscript</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># filter your gtf for the gene of interest, here "SOD1"</span></span>
<span><span class="va">gene_of_interest</span> <span class="op">&lt;-</span> <span class="st">"SOD1"</span></span>
<span></span>
<span><span class="va">sod1_annotation_from_gtf</span> <span class="op">&lt;-</span> <span class="va">gtf</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span>, </span>
<span>    <span class="va">gene_name</span> <span class="op">==</span> <span class="va">gene_of_interest</span></span>
<span>  <span class="op">)</span> </span>
<span></span>
<span><span class="co"># extract the required annotation columns</span></span>
<span><span class="va">sod1_annotation_from_gtf</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation_from_gtf</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>    <span class="va">seqnames</span>,</span>
<span>    <span class="va">start</span>,</span>
<span>    <span class="va">end</span>,</span>
<span>    <span class="va">strand</span>,</span>
<span>    <span class="va">type</span>,</span>
<span>    <span class="va">gene_name</span>,</span>
<span>    <span class="va">transcript_name</span>,</span>
<span>    <span class="va">transcript_biotype</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">sod1_annotation_from_gtf</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;   seqnames    start      end strand type        gene_name transcript_name</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">668</span>931 +      gene        SOD1      <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">668</span>931 +      transcript  SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">659</span>784 +      exon        SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       31<span style="text-decoration: underline;">659</span>770 31<span style="text-decoration: underline;">659</span>784 +      CDS         SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       31<span style="text-decoration: underline;">659</span>770 31<span style="text-decoration: underline;">659</span>772 +      start_codon SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       31<span style="text-decoration: underline;">663</span>790 31<span style="text-decoration: underline;">663</span>886 +      exon        SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="importing-data-from-a-bed-file">Importing data from a bed file<a class="anchor" aria-label="anchor" href="#importing-data-from-a-bed-file"></a>
</h3>
<p>If users would like to plot ranges from a <code>bed</code> file using
<code>ggtranscript</code>, they can first import the <code>bed</code>
file into <code>R</code> using <code><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html" class="external-link">rtracklayer::import.bed()</a></code>.
This method will create a <code>UCSCData</code> object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># for the example, we'll use the test bed file provided by rtracklayer </span></span>
<span><span class="va">test_bed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"tests/test.bed"</span>, package <span class="op">=</span> <span class="st">"rtracklayer"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bed</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html" class="external-link">import.bed</a></span><span class="op">(</span><span class="va">test_bed</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "UCSCData"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "rtracklayer"</span></span></code></pre></div>
<p>A <code>UCSCData</code> object can be coerced into a
<code>tibble</code>, a data structure which can be plotted using
<code>ggplot2</code>/<code>ggtranscript</code>, using
<code><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">dplyr::as_tibble()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">bed</span> <span class="op">&lt;-</span> <span class="va">bed</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "tbl_df"     "tbl"        "data.frame"</span></span>
<span></span>
<span><span class="va">bed</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 12</span></span></span>
<span><span class="co">#&gt;   seqnames   start    end width strand name  score itemRgb thick.start thick.end</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr7      1.27<span style="color: #949494;">e</span>8 1.27<span style="color: #949494;">e</span>8  <span style="text-decoration: underline;">1</span>167 +      Pos1      0 #FF0000   127<span style="text-decoration: underline;">471</span>197 127<span style="text-decoration: underline;">472</span>363</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr7      1.27<span style="color: #949494;">e</span>8 1.27<span style="color: #949494;">e</span>8  <span style="text-decoration: underline;">1</span>167 +      Pos2      2 #FF0000   127<span style="text-decoration: underline;">472</span>364 127<span style="text-decoration: underline;">473</span>530</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr7      1.27<span style="color: #949494;">e</span>8 1.27<span style="color: #949494;">e</span>8  <span style="text-decoration: underline;">1</span>167 -      Neg1      0 #FF0000   127<span style="text-decoration: underline;">473</span>531 127<span style="text-decoration: underline;">474</span>697</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr9      1.27<span style="color: #949494;">e</span>8 1.27<span style="color: #949494;">e</span>8  <span style="text-decoration: underline;">1</span>167 +      Pos3      5 #FF0000   127<span style="text-decoration: underline;">474</span>698 127<span style="text-decoration: underline;">475</span>864</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chr9      1.27<span style="color: #949494;">e</span>8 1.27<span style="color: #949494;">e</span>8  <span style="text-decoration: underline;">1</span>167 -      Neg2      5 #0000FF   127<span style="text-decoration: underline;">475</span>865 127<span style="text-decoration: underline;">477</span>031</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: thick.width &lt;int&gt;, blocks &lt;list&gt;</span></span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="section level3">
<h3 id="required-aesthetics">Required aesthetics<a class="anchor" aria-label="anchor" href="#required-aesthetics"></a>
</h3>
<p><code>ggtranscript</code> introduces 5 new geoms designed to simplify
the visualization of transcript structure and annotation;
<code><a href="../reference/geom_range.html">geom_range()</a></code>, <code><a href="../reference/geom_range.html">geom_half_range()</a></code>,
<code><a href="../reference/geom_intron.html">geom_intron()</a></code>, <code><a href="../reference/geom_junction.html">geom_junction()</a></code> and
<code><a href="../reference/geom_junction_label_repel.html">geom_junction_label_repel()</a></code>. The required aesthetics
(<code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>) for these geoms are designed to match the data
formats which are widely used in genetic and transcriptomic
analyses:</p>
<table class="table">
<colgroup>
<col width="11%">
<col width="15%">
<col width="48%">
<col width="24%">
</colgroup>
<thead><tr class="header">
<th align="left">Required aes()</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Required by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">xstart</td>
<td align="left">integer</td>
<td align="left">Start position in base pairs</td>
<td align="left">All geoms</td>
</tr>
<tr class="even">
<td align="left">xend</td>
<td align="left">integer</td>
<td align="left">End position in base pairs</td>
<td align="left">All geoms</td>
</tr>
<tr class="odd">
<td align="left">y</td>
<td align="left">charactor or factor</td>
<td align="left">The group used for the y axis, most often a transcript
id or name</td>
<td align="left">All geoms</td>
</tr>
<tr class="even">
<td align="left">label</td>
<td align="left">integer or charactor</td>
<td align="left">Variable used to label junction curves</td>
<td align="left">Only geom_junction_label_repel()</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="plotting_exons_and_introns">Plotting exons and introns<a class="anchor" aria-label="anchor" href="#plotting_exons_and_introns"></a>
</h3>
<p>In the simplest case, the core components of transcript structure are
exons and introns, which can be plotted using <code><a href="../reference/geom_range.html">geom_range()</a></code>
and <code><a href="../reference/geom_intron.html">geom_intron()</a></code>. In order to facilitate this,
<code>ggtranscript</code> also provides <code><a href="../reference/to_intron.html">to_intron()</a></code>, which
converts exon co-ordinates into introns. Therefore, you can plot
transcript structures with only exons as input and just a few lines of
code.</p>
<blockquote>
<p>📌: As <code>ggtranscript</code> geoms share required aesthetics, it
is recommended to set these via <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>, rather than in
the individual <code>geom_*()</code> calls to avoid code
duplication.</p>
</blockquote>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># to illustrate the package's functionality</span></span>
<span><span class="co"># ggtranscript includes example transcript annotation</span></span>
<span><span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;   seqnames    start      end strand type        gene_name transcript_name</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">668</span>931 +      gene        SOD1      <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">668</span>931 +      transcript  SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       31<span style="text-decoration: underline;">659</span>666 31<span style="text-decoration: underline;">659</span>784 +      exon        SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       31<span style="text-decoration: underline;">659</span>770 31<span style="text-decoration: underline;">659</span>784 +      CDS         SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       31<span style="text-decoration: underline;">659</span>770 31<span style="text-decoration: underline;">659</span>772 +      start_codon SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       31<span style="text-decoration: underline;">663</span>790 31<span style="text-decoration: underline;">663</span>886 +      exon        SOD1      SOD1-202       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: transcript_biotype &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="co"># extract exons</span></span>
<span><span class="va">sod1_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-range-intron-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="differentiating-utrs-from-the-coding-sequence">Differentiating UTRs from the coding sequence<a class="anchor" aria-label="anchor" href="#differentiating-utrs-from-the-coding-sequence"></a>
</h3>
<p>As suggested by it’s name, <code><a href="../reference/geom_range.html">geom_range()</a></code> is designed to
visualize range-based transcript annotation. This includes but is not
limited to exons. For instance, for protein coding transcripts it can be
useful to visually distinguish the coding sequence (CDS) of a transcript
from it’s UTRs. This can be achieved by adjusting the height and fill of
<code><a href="../reference/geom_range.html">geom_range()</a></code> and overlaying the CDS on top of the exons
(including UTRs).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># filter for only exons from protein coding transcripts</span></span>
<span><span class="va">sod1_exons_prot_cod</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># obtain cds</span></span>
<span><span class="va">sod1_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sod1_exons_prot_cod</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>        fill <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>        height <span class="op">=</span> <span class="fl">0.25</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="va">sod1_cds</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons_prot_cod</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>,</span>
<span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span>,</span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-range-intron-w-cds-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="plotting-junctions">Plotting junctions<a class="anchor" aria-label="anchor" href="#plotting-junctions"></a>
</h3>
<p><code><a href="../reference/geom_junction.html">geom_junction()</a></code> plots curved lines that are intended to
represent junction reads. Junctions are reads obtained through
RNA-sequencing (RNA-seq) data that map with gapped alignment to the
genome. Often, this gap is indicative of a splicing event, but can also
originate from other genomic events such as indels.</p>
<p>It can be useful to visually overlay junctions on top of an existing
transcript structure. For example, this can help to understand which
existing transcripts are expressed in the RNA-seq sample or inform the
location or interpretation of the novel splice sites.</p>
<p><code><a href="../reference/geom_junction_label_repel.html">geom_junction_label_repel()</a></code> adds labels to junction
curves. This can useful for labeling junctions with a measure of their
expression or support such as read counts or percent-spliced-in.
Alternatively, you may choose to visually map this measure to the
thickness of the junction curves by adjusting the the size
<code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>. Or, as shown below, both of these options can be
combined.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># extract exons and cds for the MANE-select transcript</span></span>
<span><span class="va">sod1_201_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span></span>
<span><span class="va">sod1_201_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add transcript name column to junctions for plotting</span></span>
<span><span class="va">sod1_junctions</span> <span class="op">&lt;-</span> <span class="va">sod1_junctions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>transcript_name <span class="op">=</span> <span class="st">"SOD1-201"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sod1_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="st">"white"</span>, </span>
<span>    height <span class="op">=</span> <span class="fl">0.25</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">sod1_201_cds</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/geom_junction.html">geom_junction</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">sod1_junctions</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">mean_count</span><span class="op">)</span>,</span>
<span>    junction.y.max <span class="op">=</span> <span class="fl">0.5</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_junction_label_repel.html">geom_junction_label_repel</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">sod1_junctions</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">mean_count</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    junction.y.max <span class="op">=</span> <span class="fl">0.5</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size_continuous</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-junction-1.png" width="700"></p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-transcript-structure-differences">Visualizing transcript structure differences<a class="anchor" aria-label="anchor" href="#visualizing-transcript-structure-differences"></a>
</h2>
<div class="section level3">
<h3 id="context">Context<a class="anchor" aria-label="anchor" href="#context"></a>
</h3>
<p>One of the primary reasons for visualizing transcript structures is
to better observe the differences between them. Often this can be
achieved by simply plotting the exons and introns as shown in <a href="#plotting_exons_and_introns">basic usage</a>. However, for longer,
complex transcripts this may not be as straight forward.</p>
<p>For example, the transcripts of <em>PKNOX1</em> have relatively long
introns, which makes the comparison between transcript structures
(especially small differences in exons) more difficult.</p>
<blockquote>
<p>📌: For relatively short introns, strand arrows may overlap exons. In
such cases, the <code>arrow.min.intron.length</code> parameter of
<code><a href="../reference/geom_intron.html">geom_intron()</a></code> can be used to set the minimum intron length
for a strand arrow to be plotted.</p>
</blockquote>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># extract exons</span></span>
<span><span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>, </span>
<span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">3500</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/transcript-diff-base-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="improving-transcript-structure-visualisation-using-shorten_gaps">Improving transcript structure visualisation using
<code>shorten_gaps()</code><a class="anchor" aria-label="anchor" href="#improving-transcript-structure-visualisation-using-shorten_gaps"></a>
</h3>
<p><code>ggtranscript</code> provides the helper function
<code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code>, which reduces the size of the gaps (regions
that do not overlap an exon). <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code> then rescales
the exon and intron co-ordinates, preserving the original exon
alignment. This allows you to hone in the differences of interest, such
as the exonic structure.</p>
<blockquote>
<p>📌: The rescaled co-ordinates returned by <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code>
will not match the original genomic positions. Therefore, it is
recommended that <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code> is used for visualizations
purposes only.</p>
</blockquote>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># extract exons</span></span>
<span><span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pknox1_rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/shorten_gaps.html">shorten_gaps</a></span><span class="op">(</span></span>
<span>  exons <span class="op">=</span> <span class="va">pknox1_exons</span>, </span>
<span>  introns <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, </span>
<span>  group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># shorten_gaps() returns exons and introns all in one data.frame()</span></span>
<span><span class="co"># let's split these for plotting </span></span>
<span><span class="va">pknox1_rescaled_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> </span>
<span><span class="va">pknox1_rescaled_introns</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"intron"</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="va">pknox1_rescaled_introns</span>,</span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>, </span>
<span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/shorten-gaps-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="comparing-between-two-transcripts-using-geom_half_range">Comparing between two transcripts using
<code>geom_half_range()</code><a class="anchor" aria-label="anchor" href="#comparing-between-two-transcripts-using-geom_half_range"></a>
</h3>
<p>If you are interested in the differences between two transcripts, you
can use <code><a href="../reference/geom_range.html">geom_half_range()</a></code> whilst adjusting
<code>range.orientation</code> to plot the exons from each on the
opposite sides of the transcript structure. This can reveal small
differences in exon structure, such as those observed here at the 5’
ends of <em>PKNOX1-201</em> and <em>PKNOX1-203</em>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># extract the two transcripts to be compared</span></span>
<span><span class="va">pknox1_rescaled_201_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"PKNOX1-201"</span><span class="op">)</span></span>
<span><span class="va">pknox1_rescaled_203_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"PKNOX1-203"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pknox1_rescaled_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>        y <span class="op">=</span> <span class="st">"PKNOX1-201/203"</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_half_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_rescaled_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, </span>
<span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_half_range</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="va">pknox1_rescaled_203_exons</span>,</span>
<span>        range.orientation <span class="op">=</span> <span class="st">"top"</span>, </span>
<span>        fill <span class="op">=</span> <span class="st">"purple"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_rescaled_203_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, </span>
<span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-half-range-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="comparing-many-transcripts-to-a-single-reference-transcript-using-to_diff">Comparing many transcripts to a single reference transcript using
<code>to_diff()</code><a class="anchor" aria-label="anchor" href="#comparing-many-transcripts-to-a-single-reference-transcript-using-to_diff"></a>
</h3>
<p>Sometimes, it can be useful to visualize the differences of several
transcripts with respect to one transcript. For example, you may be
interested in how other transcripts differ in structure to the
MANE-select transcript. This exploration can reveal whether certain
important regions are missing or novel regions are added, hinting at
differences in transcript function.</p>
<p><code><a href="../reference/to_diff.html">to_diff()</a></code> is a helper function designed for this
situation. Here, we apply this to <em>PKNOX1</em>, finding the
differences between all other transcripts and the MANE-select transcript
(<em>PKNOX1-201</em>).</p>
<blockquote>
<p>📌: Although below, we apply <code><a href="../reference/to_diff.html">to_diff()</a></code> to the rescaled
exons and intron (outputted by <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code>),
<code><a href="../reference/to_diff.html">to_diff()</a></code> can also be applied to the original, unscaled
transcripts with the same effect.</p>
</blockquote>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">mane</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_201_exons</span></span>
<span></span>
<span><span class="va">not_mane</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">!=</span> <span class="st">"PKNOX1-201"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pknox1_rescaled_diffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/to_diff.html">to_diff</a></span><span class="op">(</span></span>
<span>    exons <span class="op">=</span> <span class="va">not_mane</span>,</span>
<span>    ref_exons <span class="op">=</span> <span class="va">mane</span>,</span>
<span>    group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>        xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>        y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="va">pknox1_rescaled_introns</span>,</span>
<span>        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>        data <span class="op">=</span> <span class="va">pknox1_rescaled_diffs</span>,</span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">diff_type</span><span class="op">)</span>,</span>
<span>        alpha <span class="op">=</span> <span class="fl">0.2</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/to-diff-1.png" width="700"></p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="integrating-existing-ggplot2-functionality">Integrating existing <code>ggplot2</code> functionality<a class="anchor" aria-label="anchor" href="#integrating-existing-ggplot2-functionality"></a>
</h2>
<p>As a <code>ggplot2</code> extension, <code>ggtranscript</code>
inherits <code>ggplot2</code>’s familiarity and flexibility, enabling
users to intuitively adjust aesthetics, parameters, scales etc as well
as complement <code>ggtranscript</code> geoms with existing
<code>ggplot2</code> geoms to create informative, publication-ready
plots.</p>
<p>Below is a list outlining some examples of complementing
<code>ggtranscript</code> with existing <code>ggplot2</code>
functionality that we have found useful:</p>
<ul>
<li>Adding exon annotation such as <a href="https://dzhang32.github.io/ggtranscript/reference/add_exon_number.html" class="external-link">exon
number/order</a> using <code><a href="../reference/add_exon_number.html">add_exon_number()</a></code> and
<code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text()</a></code>
</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">base_sod1_plot</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span>    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span>    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span></span>
<span>  <span class="op">)</span> </span>
<span></span>
<span><span class="va">base_sod1_plot</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu"><a href="../reference/add_exon_number.html">add_exon_number</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span>, <span class="co"># plot label at midpoint of exon</span></span>
<span>      label <span class="op">=</span> <span class="va">exon_number</span></span>
<span>    <span class="op">)</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>    nudge_y <span class="op">=</span> <span class="fl">0.4</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/exon-num-ex-1.png" width="700"></p>
<ul>
<li>Zooming in on areas of interest using <code><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian()</a></code>
or <code>ggforce::facet_zoom()</code>
</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">base_sod1_plot</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">31665500</span>, <span class="fl">31669000</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/zoom-ex-1.png" width="700"></p>
<ul>
<li>Plotting mutations using <code><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline()</a></code>
</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">example_mutation</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>  transcript_name <span class="op">=</span> <span class="st">"SOD1-204"</span>, </span>
<span>  position <span class="op">=</span> <span class="fl">31661600</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># xstart and xend are set here to override the default aes()</span></span>
<span><span class="va">base_sod1_plot</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">example_mutation</span>, </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      xintercept <span class="op">=</span> <span class="va">position</span>, </span>
<span>      xstart <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>      xend <span class="op">=</span> <span class="cn">NULL</span></span>
<span>      <span class="op">)</span>, </span>
<span>    linetype <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    colour <span class="op">=</span> <span class="st">"red"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/mutation-ex-1.png" width="700"></p>
<ul>
<li>Beautifying plots using themes and scales</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">base_sod1_plot</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Position"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_y_discrete</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Transcript name"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_fill_discrete</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Transcript biotype"</span>,</span>
<span>    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Processed transcript"</span>, <span class="st">"Protein-coding"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/beautify-ex-1.png" width="700"></p>
<p><br></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><summary>
Show/hide
</summary><pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-08-24</span></span>
<span><span class="co">#&gt;  pandoc   3.2 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-5     <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase                2.64.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.50.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocIO                 1.14.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.24   <span style="color: #555555;">2024-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocParallel           1.38.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.32.1    <span style="color: #555555;">2024-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  Biostrings             2.72.1    <span style="color: #555555;">2024-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  bitops                 1.0-8     <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.40      <span style="color: #555555;">2024-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.8.0     <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.1.0     <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.3     <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  codetools              0.2-20    <span style="color: #555555;">2024-03-31</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  colorspace             2.1-1     <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.3     <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  curl                   5.2.1     <span style="color: #555555;">2024-03-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.30.1    <span style="color: #555555;">2024-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.3     <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.37    <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                * 1.1.4     <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  evaluate               0.24.0    <span style="color: #555555;">2024-06-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fansi                  1.0.6     <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  farver                 2.1.2     <span style="color: #555555;">2024-05-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.2.0     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.4     <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  generics               0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.40.1    <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.12    <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.40.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.56.1    <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  ggplot2              * 3.5.1     <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                0.9.5     <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggtranscript         * 1.0.0     <span style="color: #555555;">2024-08-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">#&gt;  glue                   1.7.0     <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gtable                 0.3.5     <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  highr                  0.11      <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1   <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4     <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr                   1.4.7     <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.38.1    <span style="color: #555555;">2024-07-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.8     <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.48      <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  labeling               0.4.3     <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.22-6    <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4     <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magrittr             * 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.7-0     <span style="color: #555555;">2024-04-26</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.16.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  matrixStats            1.3.0     <span style="color: #555555;">2024-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  munsell                0.5.1     <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pillar                 1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                2.1.0     <span style="color: #555555;">2024-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.3.2     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.13    <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.16 <span style="color: #555555;">2024-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  restfulr               0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  rjson                  0.2.22    <span style="color: #555555;">2024-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.4     <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.28      <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools              2.20.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  rtracklayer          * 1.64.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.4.1     <span style="color: #555555;">2024-05-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.42.1    <span style="color: #555555;">2024-07-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.9     <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scales                 1.3.0     <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SparseArray            1.4.8     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.34.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.1.0     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.4.0     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tibble                 3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1     <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils             1.0.0     <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  utf8                   1.2.4     <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5     <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  withr                  3.0.1     <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.47      <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XML                    3.99-0.17 <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.44.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.10    <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.50.0    <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</details>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
