<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="ggtranscript">
<title>Getting started • ggtranscript</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started">
<meta property="og:description" content="ggtranscript">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/ggtranscript.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Getting started</h1>
                        <h4 data-toc-skip class="author">David
Zhang</h4>
            <address class="author_afil">
      UCL<br><a class="author_email" href="mailto:#"></a><a href="mailto:dyzhang32@gmail.com" class="email">dyzhang32@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/vignettes/ggtranscript.Rmd" class="external-link"><code>vignettes/ggtranscript.Rmd</code></a></small>
      <div class="d-none name"><code>ggtranscript.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dzhang32/ggtranscript" class="external-link">ggtranscript</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></code></pre></div>
<p><code>ggtranscript</code> is designed to make it easy to visualize
transcript structure and annotation using <code>ggplot2</code>.</p>
<p>As the intended users are those who work with genetic and/or
transcriptomic data in <code>R</code>, this tutorial assumes a basic
understanding of transcript annotation and familiarity with
<code>ggplot2</code>.</p>
<p><br></p>
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="section level3">
<h3 id="required-aesthetics">Required aesthetics<a class="anchor" aria-label="anchor" href="#required-aesthetics"></a>
</h3>
<p><code>ggtranscript</code> introduces 5 new geoms designed to simplify
the visualization of transcript structure and annotation;
<code><a href="../reference/geom_range.html">geom_range()</a></code>, <code><a href="../reference/geom_range.html">geom_half_range()</a></code>,
<code><a href="../reference/geom_intron.html">geom_intron()</a></code>, <code><a href="../reference/geom_junction.html">geom_junction()</a></code> and
<code><a href="../reference/geom_junction_label_repel.html">geom_junction_label_repel()</a></code>. The required aesthetics
(<code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>) for these geoms are designed to match the data
formats which are widely used in genetic and transcriptomic
analyses:</p>
<table class="table">
<colgroup>
<col width="11%">
<col width="15%">
<col width="48%">
<col width="24%">
</colgroup>
<thead><tr class="header">
<th align="left">Required aes()</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Required by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">xstart</td>
<td align="left">integer</td>
<td align="left">Start position in base pairs</td>
<td align="left">All geoms</td>
</tr>
<tr class="even">
<td align="left">xend</td>
<td align="left">integer</td>
<td align="left">End position in base pairs</td>
<td align="left">All geoms</td>
</tr>
<tr class="odd">
<td align="left">y</td>
<td align="left">charactor or factor</td>
<td align="left">The group used for the y axis, most often a transcript
id or name</td>
<td align="left">All geoms</td>
</tr>
<tr class="even">
<td align="left">label</td>
<td align="left">integer or charactor</td>
<td align="left">Variable used to label junction curves</td>
<td align="left">Only geom_junction_label_repel()</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>In order to showcase the package’s functionality,
<code>ggtranscript</code> includes example transcript annotation for the
genes <em>SOD1</em> and <em>PKNOX1</em>, as well as a set of unannotated
junctions associated with <em>SOD1</em>. These specific genes are
unimportant, chosen arbitrarily for illustration. However, it worth
noting that the input data for <code>ggtranscript</code>, as a
<code>ggplot2</code> extension, is required be a <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/data.frame" class="external-link"><code>data.frame</code></a>
or <a href="https://tibble.tidyverse.org" class="external-link"><code>tibble</code></a>.</p>
<p>You may be asking, what if I have <code>GenomicRanges</code> object?
Almost always, you can easily transform <code>GenomicRanges</code> into
a <code>data.frame</code> using a helper function such as
<code><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame()</a></code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="co">#&gt;   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      gene  SOD1      <span style="color: #BB0000;">NA</span>              <span style="color: #BB0000;">NA</span>              </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      tran… SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      CDS   SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      star… SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding</span>

<span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="co">#&gt;   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      gene  PKNOX1    <span style="color: #BB0000;">NA</span>              <span style="color: #BB0000;">NA</span>              </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      tran… PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       4.30<span style="color: #949494;">e</span>7 4.30<span style="color: #949494;">e</span>7 +      exon  PKNOX1    PKNOX1-203      protein_coding</span>

<span class="va">sod1_junctions</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 5</span></span>
<span class="co">#&gt;   seqnames    start      end strand mean_count</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr21    31<span style="text-decoration: underline;">659</span>787 31<span style="text-decoration: underline;">666</span>448 +           0.463</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">660</span>554 +           0.831</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">663</span>794 +           0.316</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr21    31<span style="text-decoration: underline;">659</span>842 31<span style="text-decoration: underline;">667</span>257 +           4.35 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chr21    31<span style="text-decoration: underline;">660</span>351 31<span style="text-decoration: underline;">663</span>789 +           0.324</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plotting_exons_and_introns">Plotting exons and introns<a class="anchor" aria-label="anchor" href="#plotting_exons_and_introns"></a>
</h3>
<p>In the simplest case, the core components of transcript structure are
exons and introns, which can be plotted using <code><a href="../reference/geom_range.html">geom_range()</a></code>
and <code><a href="../reference/geom_intron.html">geom_intron()</a></code>. In order to facilitate this,
<code>ggtranscript</code> also provides <code><a href="../reference/to_intron.html">to_intron()</a></code>, which
converts exon co-ordinates into introns. Therefore, you can plot
transcript structures with only exons as input and just a few lines of
code.</p>
<blockquote>
<p>📌: As <code>ggtranscript</code> geoms share required aesthetics, it
is recommended to set these via <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>, rather than in
the individual <code>geom_*()</code> calls to avoid code
duplication.</p>
</blockquote>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># to illustrate the package's functionality</span>
<span class="co"># ggtranscript includes example transcript annotation</span>
<span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="co">#&gt;   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      gene  SOD1      <span style="color: #BB0000;">NA</span>              <span style="color: #BB0000;">NA</span>              </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      tran… SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      CDS   SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      star… SOD1      SOD1-202        protein_coding  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 21       3.17<span style="color: #949494;">e</span>7 3.17<span style="color: #949494;">e</span>7 +      exon  SOD1      SOD1-202        protein_coding</span>

<span class="co"># extract exons</span>
<span class="va">sod1_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span>

<span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-range-intron-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="differentiating-utrs-from-the-coding-sequence">Differentiating UTRs from the coding sequence<a class="anchor" aria-label="anchor" href="#differentiating-utrs-from-the-coding-sequence"></a>
</h3>
<p>As suggested by it’s name, <code><a href="../reference/geom_range.html">geom_range()</a></code> is designed to
visualize range-based transcript annotation. This includes but is not
limited to exons. For instance, for protein coding transcripts it can be
useful to visually distinguish the coding sequence (CDS) of a transcript
from it’s UTRs. This can be achieved by adjusting the height and fill of
<code><a href="../reference/geom_range.html">geom_range()</a></code> and overlaying the CDS on top of the exons
(including UTRs).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># filter for only exons from protein coding transcripts</span>
<span class="va">sod1_exons_prot_cod</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span>

<span class="co"># obtain cds</span>
<span class="va">sod1_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span>

<span class="va">sod1_exons_prot_cod</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        fill <span class="op">=</span> <span class="st">"white"</span>,
        height <span class="op">=</span> <span class="fl">0.25</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">sod1_cds</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_exons_prot_cod</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>,
        arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span>,
    <span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-range-intron-w-cds-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="plotting-junctions">Plotting junctions<a class="anchor" aria-label="anchor" href="#plotting-junctions"></a>
</h3>
<p><code><a href="../reference/geom_junction.html">geom_junction()</a></code> plots curved lines that are intended to
represent junction reads. Junctions are reads obtained through
RNA-sequencing (RNA-seq) data that map with gapped alignment to the
genome. Often, this gap is indicative of a splicing event, but can also
originate from other genomic events such as indels.</p>
<p>It can be useful to visually overlay junctions on top of an existing
transcript structure. For example, this can help to understand which
existing transcripts are expressed in the RNA-seq sample or inform the
location or interpretation of the novel splice sites.</p>
<p><code><a href="../reference/geom_junction_label_repel.html">geom_junction_label_repel()</a></code> adds labels to junction
curves. This can useful for labeling junctions with a measure of their
expression or support such as read counts or percent-spliced-in.
Alternatively, you may choose to visually map this measure to the
thickness of the junction curves by adjusting the the size
<code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>. Or, as shown below, both of these options can be
combined.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># extract exons and cds for the MANE-select transcript</span>
<span class="va">sod1_201_exons</span> <span class="op">&lt;-</span> <span class="va">sod1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span>
<span class="va">sod1_201_cds</span> <span class="op">&lt;-</span> <span class="va">sod1_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"SOD1-201"</span><span class="op">)</span>

<span class="co"># add transcript name column to junctions for plotting</span>
<span class="va">sod1_junctions</span> <span class="op">&lt;-</span> <span class="va">sod1_junctions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>transcript_name <span class="op">=</span> <span class="st">"SOD1-201"</span><span class="op">)</span>

<span class="va">sod1_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
    xstart <span class="op">=</span> <span class="va">start</span>,
    xend <span class="op">=</span> <span class="va">end</span>,
    y <span class="op">=</span> <span class="va">transcript_name</span>
  <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
    fill <span class="op">=</span> <span class="st">"white"</span>, 
    height <span class="op">=</span> <span class="fl">0.25</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_201_cds</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">sod1_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/geom_junction.html">geom_junction</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_junctions</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">mean_count</span><span class="op">)</span>,
    junction.y.max <span class="op">=</span> <span class="fl">0.5</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_junction_label_repel.html">geom_junction_label_repel</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">sod1_junctions</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">mean_count</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
    junction.y.max <span class="op">=</span> <span class="fl">0.5</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size_continuous</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-junction-1.png" width="700"></p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-transcript-structure-differences">Visualizing transcript structure differences<a class="anchor" aria-label="anchor" href="#visualizing-transcript-structure-differences"></a>
</h2>
<div class="section level3">
<h3 id="context">Context<a class="anchor" aria-label="anchor" href="#context"></a>
</h3>
<p>One of the primary reasons for visualizing transcript structures is
to better observe the differences between them. Often this can be
achieved by simply plotting the exons and introns as shown in <a href="#plotting_exons_and_introns">basic usage</a>. However, for longer,
complex transcripts this may not be as straight forward.</p>
<p>For example, the transcripts of <em>PKNOX1</em> have relatively long
introns, which makes the comparison between transcript structures
(especially small differences in exons) more difficult. &gt; 📌: For
relatively short introns, strand arrows may overlap exons. In such
cases, the <code>arrow.min.intron.length</code> parameter of
<code><a href="../reference/geom_intron.html">geom_intron()</a></code> can be used to set the minimum intron length
for a strand arrow to be plotted.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># extract exons</span>
<span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span>

<span class="va">pknox1_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>, 
        arrow.min.intron.length <span class="op">=</span> <span class="fl">3500</span>
    <span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/transcript-diff-base-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="improving-transcript-structure-visualisation-using-shorten_gaps">Improving transcript structure visualisation using
<code>shorten_gaps()</code><a class="anchor" aria-label="anchor" href="#improving-transcript-structure-visualisation-using-shorten_gaps"></a>
</h3>
<p><code>ggtranscript</code> provides the helper function
<code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code>, which reduces the size of the gaps (regions
that do not overlap an intron). <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code> then
rescales the exon and intron co-ordinates, preserving the original exon
alignment. This allows you to hone in the differences of interest, such
as the exonic structure.</p>
<blockquote>
<p>📌: The rescaled co-ordinates returned by <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code>
will not match the original genomic positions. Therefore, it is
recommended that <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code> is used for visualizations
purposes only.</p>
</blockquote>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># extract exons</span>
<span class="va">pknox1_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_annotation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span>

<span class="va">pknox1_rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/shorten_gaps.html">shorten_gaps</a></span><span class="op">(</span>
  exons <span class="op">=</span> <span class="va">pknox1_exons</span>, 
  introns <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, 
  group_var <span class="op">=</span> <span class="st">"transcript_name"</span>
<span class="op">)</span>

<span class="co"># shorten_gaps() returns exons and introns all in one data.frame()</span>
<span class="co"># let's split these for plotting </span>
<span class="va">pknox1_rescaled_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> 
<span class="va">pknox1_rescaled_introns</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"intron"</span><span class="op">)</span> 

<span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">pknox1_rescaled_introns</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>, 
        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span>
    <span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/shorten-gaps-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="comparing-between-two-transcripts-using-geom_half_range">Comparing between two transcripts using
<code>geom_half_range()</code><a class="anchor" aria-label="anchor" href="#comparing-between-two-transcripts-using-geom_half_range"></a>
</h3>
<p>If you are interested in the differences between two transcripts, you
can use <code><a href="../reference/geom_range.html">geom_half_range()</a></code> whilst adjusting
<code>range.orientation</code> to plot the exons from each on the
opposite sides of the transcript structure. This can reveal small
differences in exon structure, such as those observed here at the 5’
ends of <em>PKNOX1-201</em> and <em>PKNOX1-203</em>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># extract the two transcripts to be compared</span>
<span class="va">pknox1_rescaled_201_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"PKNOX1-201"</span><span class="op">)</span>
<span class="va">pknox1_rescaled_203_exons</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"PKNOX1-203"</span><span class="op">)</span>

<span class="va">pknox1_rescaled_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="st">"PKNOX1-201/203"</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_half_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_rescaled_201_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, 
        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_half_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">pknox1_rescaled_203_exons</span>,
        range.orientation <span class="op">=</span> <span class="st">"top"</span>, 
        fill <span class="op">=</span> <span class="st">"purple"</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="../reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">pknox1_rescaled_203_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>, 
        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span>
    <span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/geom-half-range-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="comparing-many-transcripts-to-a-single-reference-transcript-using-to_diff">Comparing many transcripts to a single reference transcript using
<code>to_diff()</code><a class="anchor" aria-label="anchor" href="#comparing-many-transcripts-to-a-single-reference-transcript-using-to_diff"></a>
</h3>
<p>Sometimes, it can be useful to visualize the differences of several
transcripts with respect to one transcript. For example, you may be
interested in how other transcripts differ in structure to the
MANE-select transcript. This exploration can reveal whether certain
important regions are missing or novel regions are added, hinting at
differences in transcript function.</p>
<p><code><a href="../reference/to_diff.html">to_diff()</a></code> is a helper function designed for this
situation. Here, we apply this to <em>PKNOX1</em>, finding the
differences between all other transcripts and the MANE-select transcript
(<em>PKNOX1-201</em>).</p>
<blockquote>
<p>📌: Although below, we apply <code><a href="../reference/to_diff.html">to_diff()</a></code> to the rescaled
exons and intron (outputted by <code><a href="../reference/shorten_gaps.html">shorten_gaps()</a></code>),
<code><a href="../reference/to_diff.html">to_diff()</a></code> can also be applied to the original, unscaled
transcripts with the same effect.</p>
</blockquote>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mane</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_201_exons</span>

<span class="va">not_mane</span> <span class="op">&lt;-</span> <span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">!=</span> <span class="st">"PKNOX1-201"</span><span class="op">)</span>

<span class="va">pknox1_rescaled_diffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/to_diff.html">to_diff</a></span><span class="op">(</span>
    exons <span class="op">=</span> <span class="va">not_mane</span>,
    ref_exons <span class="op">=</span> <span class="va">mane</span>,
    group_var <span class="op">=</span> <span class="st">"transcript_name"</span>
<span class="op">)</span>

<span class="va">pknox1_rescaled_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">pknox1_rescaled_introns</span>,
        arrow.min.intron.length <span class="op">=</span> <span class="fl">300</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="../reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">pknox1_rescaled_diffs</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">diff_type</span><span class="op">)</span>,
        alpha <span class="op">=</span> <span class="fl">0.2</span>
    <span class="op">)</span></code></pre></div>
<p><img src="ggtranscript_files/figure-html/to-diff-1.png" width="700"></p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="integrating-existing-ggplot2-functionality">Integrating existing <code>ggplot2</code> functionality<a class="anchor" aria-label="anchor" href="#integrating-existing-ggplot2-functionality"></a>
</h2>
<p>As a <code>ggplot2</code> extension, <code>ggtranscript</code>
inherits <code>ggplot2</code>’s familiarity and flexibility, enabling
users to intuitively adjust aesthetics, parameters, scales etc as well
as complement <code>ggtranscript</code> geoms with existing
<code>ggplot2</code> geoms to create informative, publication-ready
plots.</p>
<p>Below is a list outlining some examples of complementing
<code>ggtranscript</code> with existing <code>ggplot2</code>
functionality that we have found useful:</p>
<ul>
<li>Adding exon annotation such as <a href="https://dzhang32.github.io/ggtranscript/reference/add_exon_number.html" class="external-link">exon
number/order</a> using <code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text()</a></code>
</li>
<li>Zooming in on areas of interest using <code><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian()</a></code>
or <code>ggforce::facet_zoom()</code>
</li>
<li>Plotting mutations using <code><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point()</a></code>
</li>
<li>Plotting coverage using <code><a href="https://ggplot2.tidyverse.org/reference/geom_polygon.html" class="external-link">geom_polygon()</a></code>
</li>
<li>Beautifying plots using themes and scales</li>
</ul>
<p><br></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><summary>
Show/hide
</summary><pre><code><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span>
<span class="co">#&gt;  version  R Under development (unstable) (2022-03-14 r81896)</span>
<span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt;  system   x86_64, linux-gnu</span>
<span class="co">#&gt;  ui       X11</span>
<span class="co">#&gt;  language en</span>
<span class="co">#&gt;  collate  en_US.UTF-8</span>
<span class="co">#&gt;  ctype    en_US.UTF-8</span>
<span class="co">#&gt;  tz       UTC</span>
<span class="co">#&gt;  date     2022-04-24</span>
<span class="co">#&gt;  pandoc   2.17.1.1 @ /usr/local/bin/ (via rmarkdown)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span>
<span class="co">#&gt;  BiocGenerics       0.41.2   <span style="color: #555555;">2021-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocManager        1.30.17  <span style="color: #555555;">2022-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  BiocStyle        * 2.23.1   <span style="color: #555555;">2021-12-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  bitops             1.0-7    <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  bookdown           0.26     <span style="color: #555555;">2022-04-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  bslib              0.3.1    <span style="color: #555555;">2021-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  cachem             1.0.6    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  cli                3.2.0    <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  colorspace         2.0-3    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  crayon             1.5.1    <span style="color: #555555;">2022-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  desc               1.4.1    <span style="color: #555555;">2022-03-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  digest             0.6.29   <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  dplyr              1.0.8    <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  ellipsis           0.3.2    <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  evaluate           0.15     <span style="color: #555555;">2022-02-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  fansi              1.0.3    <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  farver             2.1.0    <span style="color: #555555;">2021-02-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  fastmap            1.1.0    <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  fs                 1.5.2    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  generics           0.1.2    <span style="color: #555555;">2022-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  GenomeInfoDb       1.31.10  <span style="color: #555555;">2022-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  GenomeInfoDbData   1.2.8    <span style="color: #555555;">2022-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  GenomicRanges      1.47.6   <span style="color: #555555;">2022-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  ggplot2          * 3.3.5    <span style="color: #555555;">2021-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  ggrepel            0.9.1    <span style="color: #555555;">2021-01-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  ggtranscript     * 0.99.6   <span style="color: #555555;">2022-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  glue               1.6.2    <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  gtable             0.3.0    <span style="color: #555555;">2019-03-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  highr              0.9      <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  htmltools          0.5.2    <span style="color: #555555;">2021-08-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  IRanges            2.29.1   <span style="color: #555555;">2021-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  jquerylib          0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  jsonlite           1.8.0    <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  knitr              1.38     <span style="color: #555555;">2022-03-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  labeling           0.4.2    <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  lifecycle          1.0.1    <span style="color: #555555;">2021-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  magrittr         * 2.0.3    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  memoise            2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  munsell            0.5.0    <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  pillar             1.7.0    <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  pkgconfig          2.0.3    <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  pkgdown            2.0.2    <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  purrr              0.3.4    <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  R6                 2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  ragg               1.2.2    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  Rcpp               1.0.8.3  <span style="color: #555555;">2022-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  RCurl              1.98-1.6 <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rlang              1.0.2    <span style="color: #555555;">2022-03-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rmarkdown          2.13     <span style="color: #555555;">2022-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rprojroot          2.0.3    <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  S4Vectors          0.33.17  <span style="color: #555555;">2022-04-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  sass               0.4.1    <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  scales             1.2.0    <span style="color: #555555;">2022-04-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  sessioninfo      * 1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  stringi            1.7.6    <span style="color: #555555;">2021-11-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  stringr            1.4.0    <span style="color: #555555;">2019-02-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  systemfonts        1.0.4    <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  textshaping        0.3.6    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  tibble             3.1.6    <span style="color: #555555;">2021-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  tidyselect         1.1.2    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  utf8               1.2.2    <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  vctrs              0.4.1    <span style="color: #555555;">2022-04-13</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  withr              2.5.0    <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  xfun               0.30     <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  XVector            0.35.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  yaml               2.3.5    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  zlibbioc           1.41.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</details>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
